% Generated by Sphinx.
\def\sphinxdocclass{report}
\documentclass[letterpaper,10pt,english]{sphinxmanual}
\usepackage[utf8]{inputenc}
\DeclareUnicodeCharacter{00A0}{\nobreakspace}
\usepackage[T1]{fontenc}
\usepackage{babel}
\usepackage{times}
\usepackage[Bjarne]{fncychap}
\usepackage{longtable}
\usepackage{sphinx}


\title{ExperimentDB Documentation}
\date{January 14, 2011}
\release{0.2}
\author{Dave Bridges}
\newcommand{\sphinxlogo}{}
\renewcommand{\releasename}{Release}
\makeindex

\makeatletter
\def\PYG@reset{\let\PYG@it=\relax \let\PYG@bf=\relax%
    \let\PYG@ul=\relax \let\PYG@tc=\relax%
    \let\PYG@bc=\relax \let\PYG@ff=\relax}
\def\PYG@tok#1{\csname PYG@tok@#1\endcsname}
\def\PYG@toks#1+{\ifx\relax#1\empty\else%
    \PYG@tok{#1}\expandafter\PYG@toks\fi}
\def\PYG@do#1{\PYG@bc{\PYG@tc{\PYG@ul{%
    \PYG@it{\PYG@bf{\PYG@ff{#1}}}}}}}
\def\PYG#1#2{\PYG@reset\PYG@toks#1+\relax+\PYG@do{#2}}

\def\PYG@tok@gd{\def\PYG@tc##1{\textcolor[rgb]{0.63,0.00,0.00}{##1}}}
\def\PYG@tok@gu{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.50,0.00,0.50}{##1}}}
\def\PYG@tok@gt{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.25,0.82}{##1}}}
\def\PYG@tok@gs{\let\PYG@bf=\textbf}
\def\PYG@tok@gr{\def\PYG@tc##1{\textcolor[rgb]{1.00,0.00,0.00}{##1}}}
\def\PYG@tok@cm{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.25,0.50,0.56}{##1}}}
\def\PYG@tok@vg{\def\PYG@tc##1{\textcolor[rgb]{0.73,0.38,0.84}{##1}}}
\def\PYG@tok@m{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\def\PYG@tok@mh{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\def\PYG@tok@cs{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.50,0.56}{##1}}\def\PYG@bc##1{\colorbox[rgb]{1.00,0.94,0.94}{##1}}}
\def\PYG@tok@ge{\let\PYG@it=\textit}
\def\PYG@tok@vc{\def\PYG@tc##1{\textcolor[rgb]{0.73,0.38,0.84}{##1}}}
\def\PYG@tok@il{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\def\PYG@tok@go{\def\PYG@tc##1{\textcolor[rgb]{0.19,0.19,0.19}{##1}}}
\def\PYG@tok@cp{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\def\PYG@tok@gi{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.63,0.00}{##1}}}
\def\PYG@tok@gh{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.00,0.50}{##1}}}
\def\PYG@tok@ni{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.84,0.33,0.22}{##1}}}
\def\PYG@tok@nl{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.13,0.44}{##1}}}
\def\PYG@tok@nn{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.05,0.52,0.71}{##1}}}
\def\PYG@tok@no{\def\PYG@tc##1{\textcolor[rgb]{0.38,0.68,0.84}{##1}}}
\def\PYG@tok@na{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\def\PYG@tok@nb{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\def\PYG@tok@nc{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.05,0.52,0.71}{##1}}}
\def\PYG@tok@nd{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.33,0.33,0.33}{##1}}}
\def\PYG@tok@ne{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\def\PYG@tok@nf{\def\PYG@tc##1{\textcolor[rgb]{0.02,0.16,0.49}{##1}}}
\def\PYG@tok@si{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.44,0.63,0.82}{##1}}}
\def\PYG@tok@s2{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\def\PYG@tok@vi{\def\PYG@tc##1{\textcolor[rgb]{0.73,0.38,0.84}{##1}}}
\def\PYG@tok@nt{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.02,0.16,0.45}{##1}}}
\def\PYG@tok@nv{\def\PYG@tc##1{\textcolor[rgb]{0.73,0.38,0.84}{##1}}}
\def\PYG@tok@s1{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\def\PYG@tok@gp{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.78,0.36,0.04}{##1}}}
\def\PYG@tok@sh{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\def\PYG@tok@ow{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\def\PYG@tok@sx{\def\PYG@tc##1{\textcolor[rgb]{0.78,0.36,0.04}{##1}}}
\def\PYG@tok@bp{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\def\PYG@tok@c1{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.25,0.50,0.56}{##1}}}
\def\PYG@tok@kc{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\def\PYG@tok@c{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.25,0.50,0.56}{##1}}}
\def\PYG@tok@mf{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\def\PYG@tok@err{\def\PYG@bc##1{\fcolorbox[rgb]{1.00,0.00,0.00}{1,1,1}{##1}}}
\def\PYG@tok@kd{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\def\PYG@tok@ss{\def\PYG@tc##1{\textcolor[rgb]{0.32,0.47,0.09}{##1}}}
\def\PYG@tok@sr{\def\PYG@tc##1{\textcolor[rgb]{0.14,0.33,0.53}{##1}}}
\def\PYG@tok@mo{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\def\PYG@tok@mi{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\def\PYG@tok@kn{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\def\PYG@tok@o{\def\PYG@tc##1{\textcolor[rgb]{0.40,0.40,0.40}{##1}}}
\def\PYG@tok@kr{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\def\PYG@tok@s{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\def\PYG@tok@kp{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\def\PYG@tok@w{\def\PYG@tc##1{\textcolor[rgb]{0.73,0.73,0.73}{##1}}}
\def\PYG@tok@kt{\def\PYG@tc##1{\textcolor[rgb]{0.56,0.13,0.00}{##1}}}
\def\PYG@tok@sc{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\def\PYG@tok@sb{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\def\PYG@tok@k{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\def\PYG@tok@se{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\def\PYG@tok@sd{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}

\def\PYGZbs{\char`\\}
\def\PYGZus{\char`\_}
\def\PYGZob{\char`\{}
\def\PYGZcb{\char`\}}
\def\PYGZca{\char`\^}
\def\PYGZsh{\char`\#}
\def\PYGZpc{\char`\%}
\def\PYGZdl{\char`\$}
\def\PYGZti{\char`\~}
% for compatibility with earlier versions
\def\PYGZat{@}
\def\PYGZlb{[}
\def\PYGZrb{]}
\makeatother

\begin{document}

\maketitle
\tableofcontents
\phantomsection\label{index::doc}


Contents:



\chapter{ExperimentDB Installation}
\label{installation:experimentdb-installation}\label{installation::doc}\label{installation:welcome-to-experimentdb-s-documentation}

\section{Configuration}
\label{installation:configuration}
ExperimentDB requires both a database and a webserver to be set up.  Ideally, the database should be hosted separately from the webserver and ExperimentDB installation, but this is not necessary, as both can be used from the same server.  If you are using a remote server for the database, it is best to set up a user for this database that can only be accessed from the webserver.  If you want to set up several installations (ie for different users or different laboratories), you need separate databases and ExperimentDB installations for each.  You will also need to set up the webserver with different addresses for each installation.



\section{Software Dependencies}
\label{installation:software-dependencies}\begin{enumerate}
\item {} 
\textbf{ExperimentDB source code}.  Download from one of the following:


\end{enumerate}
\begin{enumerate}
\item {} 
\href{http://github.com/davebridges/ExperimentDB/downloads}{http://github.com/davebridges/ExperimentDB/downloads} for the current release


\item {} 
\href{http://github.com/davebridges/ExperimentDB}{http://github.com/davebridges/ExperimentDB} for the source code


\item {} 
from pypi by entering:


\begin{Verbatim}[commandchars=@\[\]]
pip install experimentdb
\end{Verbatim}

\end{enumerate}

Downloading and/or unzipping will create a directory named ExperimentDB.  You can update to the newest revision at any time either using git or downloading and re-installing the newer version.  Changing or updating software versions will not alter any saved data, but you will have to update the localsettings.py file (described below).

\begin{enumerate}
\setcounter{enumi}{1}
\item {} 
\textbf{Python}.  Requires Version 2.6, is not yet compatible with Python 3.0.  Download from \href{http://www.python.org/download}{Python}.


\item {} 
\textbf{Django}.  Download from \href{http://www.djangoproject.com/download/}{Django}.  This will be automatically installed if you installed experimentdb with pip.  This will be automatically installed if you installed experimentdb with pip.


\item {} 
\textbf{Database software}.  Typically MySQL is used, but PostgreSQL, Oracle or SQLite can also be used.  You also need to install the python driver for this database (unless you are using SQLite, which is internal to Python 2.5+).  For more information see \href{http://docs.djangoproject.com/en/dev/topics/install/database-installation}{Instructions}.


\item {} 
\textbf{Biopython Packages}.  Download and install from \href{http://biopython.org}{Biopython}. This will be automatically installed if you installed experimentdb with pip.


\item {} 
\textbf{South}.  Install using pip (\textbf{pip install south}).  This will be automatically installed if you installed experimentdb with pip.


\item {} 
\textbf{Django Ajax Select}.  Install using pip (\textbf{pip install django-ajax-selects}).  This will be automatically installed if you installed experimentdb with pip.


\item {} 
\textbf{Python Imaging Library}.  Install using pip (\textbf{pip install pil}).  Available at \href{http://www.pythonware.com/products/pil/}{PIL}.  This will be automatically installed if you installed experimentdb with pip.


\end{enumerate}


\section{Database Setup}
\label{installation:database-setup}\label{installation:pil}\begin{enumerate}
\item {} 
Create a new database.  You need to record the user, password, host and database name.  Refer to the database documentation for how to do this with a specific database engine.  If you are using SQLite3, you only need to set the engine and the database name.  It is recommended to use MySQL.


\item {} 
Go to localsettings\_empty.py and edit the settings:


\begin{Verbatim}[commandchars=@\[\]]
ENGINE: 'mysql', 'postgresql@_psycopg2' or 'sqlite3 depending on the database software used.
NAME: database name
USER: database user.  Unless using sqlite3
PASSWORD: database password.  Unless using sqlite3
HOST: database host.
\end{Verbatim}

\item {} 
Save this file as localsettings.py in the main ExperimentDB directory.


\item {} 
Run the test client by going into the experimentdb directory and running the following.  There should be no errors at this point:


\begin{Verbatim}[commandchars=@\[\]]
python manage.py test
\end{Verbatim}

\item {} 
Generate the initial database tables by entering:


\begin{Verbatim}[commandchars=@\[\]]
python manage.py syncdb
\end{Verbatim}

\item {} 
When asked generate an administrative superuser and set the email and password.


\end{enumerate}


\section{Web Server Setup}
\label{installation:web-server-setup}
You need to set up a server to serve both the django installation and saved files.  For the saved files.  I recommend using apache for both.  The preferred setup is to use Apache2 with mod\_wsgi.  The following is a httpd.conf example where the code is placed in /usr/src/django/experimentdb:


\begin{Verbatim}[commandchars=@\[\]]
Alias /static /usr/src/django/experimentdb/media
Alias /media /usr/src/django/experimentdb/media

    @textless[]Directory /usr/src/django/experimentdb/media@textgreater[]
    Order allow,deny
    Allow from all
@textless[]/Directory@textgreater[]

WSGIScriptAlias /experimentdb /usr/src/django/experimentdb/apache/django.wsgi

@textless[]Directory /usr/src/django/experimentdb/apache@textgreater[]
    Order deny,allow
    Allow from all
@textless[]/Directory@textgreater[]
\end{Verbatim}

If you want to restrict access to these files, change the Allow from all directive to specific domains or ip addresses (for example Allow from 192.168.0.0/99 would allow from 192.168.0.0 to 192.168.0.99)



\section{Final Configuration and User Setup}
\label{installation:final-configuration-and-user-setup}\begin{enumerate}
\item {} 
Go to experimentdb/admin/auth/users/ and create users, selecting usernames, full names, password (or have the user set the password) and then choose group permissions.


\end{enumerate}


\chapter{Package Details}
\label{api:module-experimentdb}\label{api:package-details}\label{api::doc}\index{experimentdb (module)}
The experimentDB is a web-based application for the storage, organization and communication of experimental data with a focus on molecular biology and biochemical data. This application also stores data regarding reagents, including antibodies, constructs and other biomolecules as well as tracks the distribution of reagents. There is also some preliminary interfaces to other web resources.



\section{Data Package}
\label{api:module-experimentdb.data}\label{api:data-package}\index{experimentdb.data (module)}
This package describes experimental data.


There are three main models in this package.  These models cover experiments, results and protocols.  There are model specifications described in experimentdb.models and views described in either experimentdb.views or experimentdb.urls for either custom or generic views respectively



\subsection{Models}
\label{api:models}\label{api:module-experimentdb.data.models}\index{experimentdb.data.models (module)}\index{Experiment (class in experimentdb.data.models)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.models.Experiment}\pysiglinewithargsret{\strong{class }\code{experimentdb.data.models.}\bfcode{Experiment}}{\emph{*args}, \emph{**kwargs}}{}
Experiment(experimentID, experiment, assay, experiment\_date, comments, public, published, sample\_storage)

\index{Experiment.DoesNotExist}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.models.Experiment.DoesNotExist}\pysigline{\strong{exception }\bfcode{DoesNotExist}}{}
\end{fulllineitems}

\index{Experiment.MultipleObjectsReturned}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.models.Experiment.MultipleObjectsReturned}\pysigline{\strong{exception }\code{Experiment.}\bfcode{MultipleObjectsReturned}}{}
\end{fulllineitems}

\index{antibodies (experimentdb.data.models.Experiment attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.models.Experiment.antibodies}\pysigline{\code{Experiment.}\bfcode{antibodies}}{}
\end{fulllineitems}

\index{cellline (experimentdb.data.models.Experiment attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.models.Experiment.cellline}\pysigline{\code{Experiment.}\bfcode{cellline}}{}
\end{fulllineitems}

\index{chemicals (experimentdb.data.models.Experiment attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.models.Experiment.chemicals}\pysigline{\code{Experiment.}\bfcode{chemicals}}{}
\end{fulllineitems}

\index{constructs (experimentdb.data.models.Experiment attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.models.Experiment.constructs}\pysigline{\code{Experiment.}\bfcode{constructs}}{}
\end{fulllineitems}

\index{get\_absolute\_url() (experimentdb.data.models.Experiment method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.models.Experiment.get_absolute_url}\pysiglinewithargsret{\code{Experiment.}\bfcode{get\_absolute\_url}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{get\_next\_by\_experiment\_date() (experimentdb.data.models.Experiment method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.models.Experiment.get_next_by_experiment_date}\pysiglinewithargsret{\code{Experiment.}\bfcode{get\_next\_by\_experiment\_date}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{get\_previous\_by\_experiment\_date() (experimentdb.data.models.Experiment method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.models.Experiment.get_previous_by_experiment_date}\pysiglinewithargsret{\code{Experiment.}\bfcode{get\_previous\_by\_experiment\_date}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{project (experimentdb.data.models.Experiment attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.models.Experiment.project}\pysigline{\code{Experiment.}\bfcode{project}}{}
\end{fulllineitems}

\index{protein (experimentdb.data.models.Experiment attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.models.Experiment.protein}\pysigline{\code{Experiment.}\bfcode{protein}}{}
\end{fulllineitems}

\index{protocol (experimentdb.data.models.Experiment attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.models.Experiment.protocol}\pysigline{\code{Experiment.}\bfcode{protocol}}{}
\end{fulllineitems}

\index{researcher (experimentdb.data.models.Experiment attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.models.Experiment.researcher}\pysigline{\code{Experiment.}\bfcode{researcher}}{}
\end{fulllineitems}

\index{result\_set (experimentdb.data.models.Experiment attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.models.Experiment.result_set}\pysigline{\code{Experiment.}\bfcode{result\_set}}{}
\end{fulllineitems}

\index{siRNA (experimentdb.data.models.Experiment attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.models.Experiment.siRNA}\pysigline{\code{Experiment.}\bfcode{siRNA}}{}
\end{fulllineitems}

\index{strain (experimentdb.data.models.Experiment attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.models.Experiment.strain}\pysigline{\code{Experiment.}\bfcode{strain}}{}
\end{fulllineitems}

\index{subproject (experimentdb.data.models.Experiment attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.models.Experiment.subproject}\pysigline{\code{Experiment.}\bfcode{subproject}}{}
\end{fulllineitems}


\end{fulllineitems}

\index{Protocol (class in experimentdb.data.models)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.models.Protocol}\pysiglinewithargsret{\strong{class }\code{experimentdb.data.models.}\bfcode{Protocol}}{\emph{*args}, \emph{**kwargs}}{}
Describes the protocol or protocols used to perform each experiment.


This model stores information about the protocol used for an experiment.


An experiment may have several protocols attached to it.  For example, one could culture and transfect cells, then generate lysates then do some western blots.


Since migrating to a mediawiki based protocol storage system, the wiki\_page attribute indicates the protocol wiki page.  In this model, the \textbf{protocol\_revision} attribute indicates the particular revision of the protocol used for that particular experiment.  In this way a permalink can be generated to the specific protocol used for a particular experiment.  To find the protocol revision number, mouse over the permanent link on the protocol and record the number at the end of the url.

\index{Protocol.DoesNotExist}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.models.Protocol.DoesNotExist}\pysigline{\strong{exception }\bfcode{DoesNotExist}}{}
\end{fulllineitems}

\index{Protocol.MultipleObjectsReturned}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.models.Protocol.MultipleObjectsReturned}\pysigline{\strong{exception }\code{Protocol.}\bfcode{MultipleObjectsReturned}}{}
\end{fulllineitems}

\index{experiment\_set (experimentdb.data.models.Protocol attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.models.Protocol.experiment_set}\pysigline{\code{Protocol.}\bfcode{experiment\_set}}{}
\end{fulllineitems}

\index{get\_absolute\_url() (experimentdb.data.models.Protocol method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.models.Protocol.get_absolute_url}\pysiglinewithargsret{\code{Protocol.}\bfcode{get\_absolute\_url}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}


\end{fulllineitems}

\index{Result (class in experimentdb.data.models)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.models.Result}\pysiglinewithargsret{\strong{class }\code{experimentdb.data.models.}\bfcode{Result}}{\emph{*args}, \emph{**kwargs}}{}
Result(id, experiment\_id, conclusions, file1, file2, file3, rawscan1, rawscan2, rawscan3, rawscan4, rawscan5, result\_figure1, result\_figure2, public, published)

\index{Result.DoesNotExist}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.models.Result.DoesNotExist}\pysigline{\strong{exception }\bfcode{DoesNotExist}}{}
\end{fulllineitems}

\index{Result.MultipleObjectsReturned}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.models.Result.MultipleObjectsReturned}\pysigline{\strong{exception }\code{Result.}\bfcode{MultipleObjectsReturned}}{}
\end{fulllineitems}

\index{experiment (experimentdb.data.models.Result attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.models.Result.experiment}\pysigline{\code{Result.}\bfcode{experiment}}{}
\end{fulllineitems}

\index{get\_absolute\_url() (experimentdb.data.models.Result method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.models.Result.get_absolute_url}\pysiglinewithargsret{\code{Result.}\bfcode{get\_absolute\_url}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}


\end{fulllineitems}

\index{Sequencing (class in experimentdb.data.models)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.models.Sequencing}\pysiglinewithargsret{\strong{class }\code{experimentdb.data.models.}\bfcode{Sequencing}}{\emph{*args}, \emph{**kwargs}}{}
Sequencing(id, clone\_name, construct\_id, primer\_id, file, sequence, correct, notes, date, sample\_number, gel\_number, lane\_number)

\index{Sequencing.DoesNotExist}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.models.Sequencing.DoesNotExist}\pysigline{\strong{exception }\bfcode{DoesNotExist}}{}
\end{fulllineitems}

\index{Sequencing.MultipleObjectsReturned}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.models.Sequencing.MultipleObjectsReturned}\pysigline{\strong{exception }\code{Sequencing.}\bfcode{MultipleObjectsReturned}}{}
\end{fulllineitems}

\index{construct (experimentdb.data.models.Sequencing attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.models.Sequencing.construct}\pysigline{\code{Sequencing.}\bfcode{construct}}{}
\end{fulllineitems}

\index{get\_next\_by\_date() (experimentdb.data.models.Sequencing method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.models.Sequencing.get_next_by_date}\pysiglinewithargsret{\code{Sequencing.}\bfcode{get\_next\_by\_date}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{get\_previous\_by\_date() (experimentdb.data.models.Sequencing method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.models.Sequencing.get_previous_by_date}\pysiglinewithargsret{\code{Sequencing.}\bfcode{get\_previous\_by\_date}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{primer (experimentdb.data.models.Sequencing attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.models.Sequencing.primer}\pysigline{\code{Sequencing.}\bfcode{primer}}{}
\end{fulllineitems}


\end{fulllineitems}



\subsection{Views}
\label{api:module-experimentdb.data.views}\label{api:views}\index{experimentdb.data.views (module)}
This module provides the views for working with the data package.
This module will generate index and detail views for experiments and protocols as well as for the form to enter new results through an experiment.  Several other generic views are found in data.urls.

\index{experiment() (in module experimentdb.data.views)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.views.experiment}\pysiglinewithargsret{\code{experimentdb.data.views.}\bfcode{experiment}}{\emph{request}, \emph{*args}, \emph{**kwargs}}{}
This renders a detailed page of an experiment.


The view will show the experiment, and all associated reagents, proteins, projects and results associated with this object.


\end{fulllineitems}

\index{experiment\_edit() (in module experimentdb.data.views)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.views.experiment_edit}\pysiglinewithargsret{\code{experimentdb.data.views.}\bfcode{experiment\_edit}}{\emph{request}, \emph{*args}, \emph{**kwargs}}{}
Renders a form to edit an experiment and associated formsets for experimental results.


Takes a request in the form of experiment/(experimentID)/edit and returns the experiment\_result\_form.html form.


\end{fulllineitems}

\index{index() (in module experimentdb.data.views)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.views.index}\pysiglinewithargsret{\code{experimentdb.data.views.}\bfcode{index}}{\emph{request}, \emph{*args}, \emph{**kwargs}}{}
This view shows a list of all experiments.


This list is ordered by the experiment date in descending order.  This view could potentially be rendered by a generic view.


\end{fulllineitems}

\index{protocol\_detail() (in module experimentdb.data.views)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.views.protocol_detail}\pysiglinewithargsret{\code{experimentdb.data.views.}\bfcode{protocol\_detail}}{\emph{request}, \emph{*args}, \emph{**kwargs}}{}
This renders a view in which a protocol detail page is shown.


This view should be deprecated in favor of a redirection directly to the wiki page for this protocol


\end{fulllineitems}

\index{protocol\_list() (in module experimentdb.data.views)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.views.protocol_list}\pysiglinewithargsret{\code{experimentdb.data.views.}\bfcode{protocol\_list}}{\emph{request}, \emph{*args}, \emph{**kwargs}}{}
This renders a view in which all protocols are displayed.


In the case of deprecated protocols (i.e. protocols not marked as active), these are not shown.  This view could also be rendered as a generic view.


\end{fulllineitems}

\index{result\_new() (in module experimentdb.data.views)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.views.result_new}\pysiglinewithargsret{\code{experimentdb.data.views.}\bfcode{result\_new}}{\emph{request}, \emph{*args}, \emph{**kwargs}}{}
This renders a form to add a new result.


This view will be sent from a particular experiment and will attach the result to that particular experiment.


\end{fulllineitems}



\subsection{Lookups}
\label{api:lookups}\label{api:module-experimentdb.data.lookups}\index{experimentdb.data.lookups (module)}
This is a configuration file for the ajax lookups for the data app.


See \href{http://code.google.com/p/django-ajax-selects/}{http://code.google.com/p/django-ajax-selects/} for information about configuring the ajax lookups.

\index{ProtocolLookup (class in experimentdb.data.lookups)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.lookups.ProtocolLookup}\pysigline{\strong{class }\code{experimentdb.data.lookups.}\bfcode{ProtocolLookup}}{}
This is the generic lookup for protocols.


It is to be used for all protocol requests and directs to the `protocol' channel.

\index{format\_item() (experimentdb.data.lookups.ProtocolLookup method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.lookups.ProtocolLookup.format_item}\pysiglinewithargsret{\bfcode{format\_item}}{\emph{protocol}}{}
the display of a currently selected object in the area below the search box. html is OK


\end{fulllineitems}

\index{format\_result() (experimentdb.data.lookups.ProtocolLookup method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.lookups.ProtocolLookup.format_result}\pysiglinewithargsret{\bfcode{format\_result}}{\emph{protocol}}{}
This controls the display of the dropdown menu.


This is set to show the unicode name of the protocol.


\end{fulllineitems}

\index{get\_objects() (experimentdb.data.lookups.ProtocolLookup method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.lookups.ProtocolLookup.get_objects}\pysiglinewithargsret{\bfcode{get\_objects}}{\emph{ids}}{}
given a list of ids, return the objects ordered as you would like them on the admin page.
this is for displaying the currently selected items (in the case of a ManyToMany field)


\end{fulllineitems}

\index{get\_query() (experimentdb.data.lookups.ProtocolLookup method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.data.lookups.ProtocolLookup.get_query}\pysiglinewithargsret{\bfcode{get\_query}}{\emph{q}, \emph{request}}{}
This sets up the query for the lookup.


The lookup searches the name of the protocol.


\end{fulllineitems}


\end{fulllineitems}



\subsection{URLconfs}
\label{api:urlconfs}\label{api:module-experimentdb.data.urls}\index{experimentdb.data.urls (module)}
This package stores views for the data package.



\subsection{Tests}
\label{api:tests}

\subsection{Fixtures}
\label{api:fixtures}

\section{Datasets Package}
\label{api:datasets-package}\label{api:module-experimentdb.datasets}\index{experimentdb.datasets (module)}

\subsection{Models}
\label{api:id1}\phantomsection\label{api:module-experimentdb.datasets.models}\index{experimentdb.datasets.models (module)}\index{IL10\_TNFa\_Microarray (class in experimentdb.datasets.models)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.datasets.models.IL10_TNFa_Microarray}\pysiglinewithargsret{\strong{class }\code{experimentdb.datasets.models.}\bfcode{IL10\_TNFa\_Microarray}}{\emph{*args}, \emph{**kwargs}}{}
IL10\_TNFa\_Microarray(id, ill\_ID, Control\_1\_2008, Control\_2\_2008, Control\_1\_2009, Control\_2\_2009, Control\_3\_2009, Control\_4\_2009, TNFa\_1\_2008, TNFa\_2\_2008, TNFa\_1\_2009, TNFa\_2\_2009, TNFa\_3\_2009, TNFa\_4\_2009, Both\_1\_2008, Both\_2\_2008, Both\_1\_2009, Both\_2\_2009, Both\_3\_2009, Both\_4\_2009, IL10\_1\_2008, IL10\_2\_2008, IL10\_1\_2009, IL10\_2\_2009, IL10\_3\_2009, IL10\_4\_2009, GeneSymbol, GeneID, GeneName)

\index{IL10\_TNFa\_Microarray.DoesNotExist}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.datasets.models.IL10_TNFa_Microarray.DoesNotExist}\pysigline{\strong{exception }\bfcode{DoesNotExist}}{}
\end{fulllineitems}

\index{IL10\_TNFa\_Microarray.MultipleObjectsReturned}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.datasets.models.IL10_TNFa_Microarray.MultipleObjectsReturned}\pysigline{\strong{exception }\code{IL10\_TNFa\_Microarray.}\bfcode{MultipleObjectsReturned}}{}
\end{fulllineitems}


\end{fulllineitems}

\index{PI35P2\_Binding\_Screen\_SP (class in experimentdb.datasets.models)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.datasets.models.PI35P2_Binding_Screen_SP}\pysiglinewithargsret{\strong{class }\code{experimentdb.datasets.models.}\bfcode{PI35P2\_Binding\_Screen\_SP}}{\emph{*args}, \emph{**kwargs}}{}
PI35P2\_Binding\_Screen\_SP(id, Gene\_Name\_id, Gain\_of\_Function, Loss\_of\_Function, Candidate, Comments)

\index{PI35P2\_Binding\_Screen\_SP.DoesNotExist}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.datasets.models.PI35P2_Binding_Screen_SP.DoesNotExist}\pysigline{\strong{exception }\bfcode{DoesNotExist}}{}
\end{fulllineitems}

\index{Gene\_Name (experimentdb.datasets.models.PI35P2\_Binding\_Screen\_SP attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.datasets.models.PI35P2_Binding_Screen_SP.Gene_Name}\pysigline{\code{PI35P2\_Binding\_Screen\_SP.}\bfcode{Gene\_Name}}{}
\end{fulllineitems}

\index{PI35P2\_Binding\_Screen\_SP.MultipleObjectsReturned}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.datasets.models.PI35P2_Binding_Screen_SP.MultipleObjectsReturned}\pysigline{\strong{exception }\code{PI35P2\_Binding\_Screen\_SP.}\bfcode{MultipleObjectsReturned}}{}
\end{fulllineitems}

\index{get\_Gain\_of\_Function\_display() (experimentdb.datasets.models.PI35P2\_Binding\_Screen\_SP method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.datasets.models.PI35P2_Binding_Screen_SP.get_Gain_of_Function_display}\pysiglinewithargsret{\code{PI35P2\_Binding\_Screen\_SP.}\bfcode{get\_Gain\_of\_Function\_display}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{get\_Loss\_of\_Function\_display() (experimentdb.datasets.models.PI35P2\_Binding\_Screen\_SP method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.datasets.models.PI35P2_Binding_Screen_SP.get_Loss_of_Function_display}\pysiglinewithargsret{\code{PI35P2\_Binding\_Screen\_SP.}\bfcode{get\_Loss\_of\_Function\_display}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}


\end{fulllineitems}

\index{SGD\_GeneNames (class in experimentdb.datasets.models)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.datasets.models.SGD_GeneNames}\pysiglinewithargsret{\strong{class }\code{experimentdb.datasets.models.}\bfcode{SGD\_GeneNames}}{\emph{*args}, \emph{**kwargs}}{}
SGD\_GeneNames(Locus\_name, Other\_name, Description, Gene\_product, Phenotype, ORF\_name, SGDID)

\index{Bait\_GeneName (experimentdb.datasets.models.SGD\_GeneNames attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.datasets.models.SGD_GeneNames.Bait_GeneName}\pysigline{\bfcode{Bait\_GeneName}}{}
\end{fulllineitems}

\index{SGD\_GeneNames.DoesNotExist}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.datasets.models.SGD_GeneNames.DoesNotExist}\pysigline{\strong{exception }\bfcode{DoesNotExist}}{}
\end{fulllineitems}

\index{Hit\_GeneName (experimentdb.datasets.models.SGD\_GeneNames attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.datasets.models.SGD_GeneNames.Hit_GeneName}\pysigline{\code{SGD\_GeneNames.}\bfcode{Hit\_GeneName}}{}
\end{fulllineitems}

\index{SGD\_GeneNames.MultipleObjectsReturned}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.datasets.models.SGD_GeneNames.MultipleObjectsReturned}\pysigline{\strong{exception }\code{SGD\_GeneNames.}\bfcode{MultipleObjectsReturned}}{}
\end{fulllineitems}

\index{PI3PBP\_Gene\_Name (experimentdb.datasets.models.SGD\_GeneNames attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.datasets.models.SGD_GeneNames.PI3PBP_Gene_Name}\pysigline{\code{SGD\_GeneNames.}\bfcode{PI3PBP\_Gene\_Name}}{}
\end{fulllineitems}

\index{get\_absolute\_url() (experimentdb.datasets.models.SGD\_GeneNames method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.datasets.models.SGD_GeneNames.get_absolute_url}\pysiglinewithargsret{\code{SGD\_GeneNames.}\bfcode{get\_absolute\_url}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{sgd\_phenotypes\_set (experimentdb.datasets.models.SGD\_GeneNames attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.datasets.models.SGD_GeneNames.sgd_phenotypes_set}\pysigline{\code{SGD\_GeneNames.}\bfcode{sgd\_phenotypes\_set}}{}
\end{fulllineitems}


\end{fulllineitems}

\index{SGD\_interactions (class in experimentdb.datasets.models)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.datasets.models.SGD_interactions}\pysiglinewithargsret{\strong{class }\code{experimentdb.datasets.models.}\bfcode{SGD\_interactions}}{\emph{*args}, \emph{**kwargs}}{}
SGD\_interactions(id, Feature\_Name\_Bait, Standard\_Gene\_Name\_Bait\_id, Feature\_Name\_Hit, Standard\_Gene\_Name\_Hit\_id, Experiment\_Type, Genetic\_or\_Physical\_Interaction, Source, Manually\_Curated\_or\_High\_Throughput, Notes, Phenotype, Reference, Citation)

\index{SGD\_interactions.DoesNotExist}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.datasets.models.SGD_interactions.DoesNotExist}\pysigline{\strong{exception }\bfcode{DoesNotExist}}{}
\end{fulllineitems}

\index{SGD\_interactions.MultipleObjectsReturned}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.datasets.models.SGD_interactions.MultipleObjectsReturned}\pysigline{\strong{exception }\code{SGD\_interactions.}\bfcode{MultipleObjectsReturned}}{}
\end{fulllineitems}

\index{Standard\_Gene\_Name\_Bait (experimentdb.datasets.models.SGD\_interactions attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.datasets.models.SGD_interactions.Standard_Gene_Name_Bait}\pysigline{\code{SGD\_interactions.}\bfcode{Standard\_Gene\_Name\_Bait}}{}
\end{fulllineitems}

\index{Standard\_Gene\_Name\_Hit (experimentdb.datasets.models.SGD\_interactions attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.datasets.models.SGD_interactions.Standard_Gene_Name_Hit}\pysigline{\code{SGD\_interactions.}\bfcode{Standard\_Gene\_Name\_Hit}}{}
\end{fulllineitems}


\end{fulllineitems}

\index{SGD\_phenotypes (class in experimentdb.datasets.models)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.datasets.models.SGD_phenotypes}\pysiglinewithargsret{\strong{class }\code{experimentdb.datasets.models.}\bfcode{SGD\_phenotypes}}{\emph{*args}, \emph{**kwargs}}{}
SGD\_phenotypes(id, Feature\_Name, Feature\_Type, Gene\_Name\_id, SGDID, Reference, Experiment\_Type, Mutant\_Type, Allele, Strain\_Background, Phenotype, Chemical, Condition, Details, Reporter)

\index{SGD\_phenotypes.DoesNotExist}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.datasets.models.SGD_phenotypes.DoesNotExist}\pysigline{\strong{exception }\bfcode{DoesNotExist}}{}
\end{fulllineitems}

\index{Gene\_Name (experimentdb.datasets.models.SGD\_phenotypes attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.datasets.models.SGD_phenotypes.Gene_Name}\pysigline{\code{SGD\_phenotypes.}\bfcode{Gene\_Name}}{}
\end{fulllineitems}

\index{SGD\_phenotypes.MultipleObjectsReturned}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.datasets.models.SGD_phenotypes.MultipleObjectsReturned}\pysigline{\strong{exception }\code{SGD\_phenotypes.}\bfcode{MultipleObjectsReturned}}{}
\end{fulllineitems}


\end{fulllineitems}



\subsection{Views}
\label{api:id2}\phantomsection\label{api:module-experimentdb.datasets.views}\index{experimentdb.datasets.views (module)}\index{sgd\_gene\_detail() (in module experimentdb.datasets.views)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.datasets.views.sgd_gene_detail}\pysiglinewithargsret{\code{experimentdb.datasets.views.}\bfcode{sgd\_gene\_detail}}{\emph{request}, \emph{gene}}{}
\end{fulllineitems}



\subsection{URLconfs}
\label{api:id3}

\subsection{Tests}
\label{api:id4}

\subsection{Fixtures}
\label{api:id5}

\section{Cloning Package}
\label{api:module-experimentdb.cloning}\label{api:cloning-package}\index{experimentdb.cloning (module)}

\subsection{Models}
\label{api:id6}\phantomsection\label{api:module-experimentdb.cloning.models}\index{experimentdb.cloning.models (module)}\index{Cloning (class in experimentdb.cloning.models)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.cloning.models.Cloning}\pysiglinewithargsret{\strong{class }\code{experimentdb.cloning.models.}\bfcode{Cloning}}{\emph{*args}, \emph{**kwargs}}{}
This model stores details about the generation of new recombinant DNA molecules.

\index{Cloning.DoesNotExist}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.cloning.models.Cloning.DoesNotExist}\pysigline{\strong{exception }\bfcode{DoesNotExist}}{}
\end{fulllineitems}

\index{Cloning.MultipleObjectsReturned}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.cloning.models.Cloning.MultipleObjectsReturned}\pysigline{\strong{exception }\code{Cloning.}\bfcode{MultipleObjectsReturned}}{}
\end{fulllineitems}

\index{construct (experimentdb.cloning.models.Cloning attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.cloning.models.Cloning.construct}\pysigline{\code{Cloning.}\bfcode{construct}}{}
\end{fulllineitems}

\index{get\_absolute\_url() (experimentdb.cloning.models.Cloning method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.cloning.models.Cloning.get_absolute_url}\pysiglinewithargsret{\code{Cloning.}\bfcode{get\_absolute\_url}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{get\_cloning\_type\_display() (experimentdb.cloning.models.Cloning method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.cloning.models.Cloning.get_cloning_type_display}\pysiglinewithargsret{\code{Cloning.}\bfcode{get\_cloning\_type\_display}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{primer\_3prime (experimentdb.cloning.models.Cloning attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.cloning.models.Cloning.primer_3prime}\pysigline{\code{Cloning.}\bfcode{primer\_3prime}}{}
\end{fulllineitems}

\index{primer\_5prime (experimentdb.cloning.models.Cloning attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.cloning.models.Cloning.primer_5prime}\pysigline{\code{Cloning.}\bfcode{primer\_5prime}}{}
\end{fulllineitems}

\index{researcher (experimentdb.cloning.models.Cloning attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.cloning.models.Cloning.researcher}\pysigline{\code{Cloning.}\bfcode{researcher}}{}
\end{fulllineitems}

\index{sequencing (experimentdb.cloning.models.Cloning attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.cloning.models.Cloning.sequencing}\pysigline{\code{Cloning.}\bfcode{sequencing}}{}
\end{fulllineitems}

\index{vector (experimentdb.cloning.models.Cloning attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.cloning.models.Cloning.vector}\pysigline{\code{Cloning.}\bfcode{vector}}{}
\end{fulllineitems}


\end{fulllineitems}

\index{Mutagenesis (class in experimentdb.cloning.models)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.cloning.models.Mutagenesis}\pysiglinewithargsret{\strong{class }\code{experimentdb.cloning.models.}\bfcode{Mutagenesis}}{\emph{*args}, \emph{**kwargs}}{}
This model contains data describing the generation of muationns in clones

\index{Mutagenesis.DoesNotExist}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.cloning.models.Mutagenesis.DoesNotExist}\pysigline{\strong{exception }\bfcode{DoesNotExist}}{}
\end{fulllineitems}

\index{Mutagenesis.MultipleObjectsReturned}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.cloning.models.Mutagenesis.MultipleObjectsReturned}\pysigline{\strong{exception }\code{Mutagenesis.}\bfcode{MultipleObjectsReturned}}{}
\end{fulllineitems}

\index{antisense\_primer (experimentdb.cloning.models.Mutagenesis attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.cloning.models.Mutagenesis.antisense_primer}\pysigline{\code{Mutagenesis.}\bfcode{antisense\_primer}}{}
\end{fulllineitems}

\index{construct (experimentdb.cloning.models.Mutagenesis attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.cloning.models.Mutagenesis.construct}\pysigline{\code{Mutagenesis.}\bfcode{construct}}{}
\end{fulllineitems}

\index{get\_absolute\_url() (experimentdb.cloning.models.Mutagenesis method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.cloning.models.Mutagenesis.get_absolute_url}\pysiglinewithargsret{\code{Mutagenesis.}\bfcode{get\_absolute\_url}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{get\_next\_by\_date\_completed() (experimentdb.cloning.models.Mutagenesis method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.cloning.models.Mutagenesis.get_next_by_date_completed}\pysiglinewithargsret{\code{Mutagenesis.}\bfcode{get\_next\_by\_date\_completed}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{get\_previous\_by\_date\_completed() (experimentdb.cloning.models.Mutagenesis method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.cloning.models.Mutagenesis.get_previous_by_date_completed}\pysiglinewithargsret{\code{Mutagenesis.}\bfcode{get\_previous\_by\_date\_completed}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{protocol (experimentdb.cloning.models.Mutagenesis attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.cloning.models.Mutagenesis.protocol}\pysigline{\code{Mutagenesis.}\bfcode{protocol}}{}
\end{fulllineitems}

\index{researcher (experimentdb.cloning.models.Mutagenesis attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.cloning.models.Mutagenesis.researcher}\pysigline{\code{Mutagenesis.}\bfcode{researcher}}{}
\end{fulllineitems}

\index{sense\_primer (experimentdb.cloning.models.Mutagenesis attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.cloning.models.Mutagenesis.sense_primer}\pysigline{\code{Mutagenesis.}\bfcode{sense\_primer}}{}
\end{fulllineitems}

\index{sequencing (experimentdb.cloning.models.Mutagenesis attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.cloning.models.Mutagenesis.sequencing}\pysigline{\code{Mutagenesis.}\bfcode{sequencing}}{}
\end{fulllineitems}

\index{template (experimentdb.cloning.models.Mutagenesis attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.cloning.models.Mutagenesis.template}\pysigline{\code{Mutagenesis.}\bfcode{template}}{}
\end{fulllineitems}


\end{fulllineitems}



\subsection{Views}
\label{api:id7}\phantomsection\label{api:module-experimentdb.cloning.views}\index{experimentdb.cloning.views (module)}

\subsection{URLconfs}
\label{api:id8}\phantomsection\label{api:module-experimentdb.cloning.urls}\index{experimentdb.cloning.urls (module)}

\subsection{Tests}
\label{api:id9}

\subsection{Fixtures}
\label{api:id10}

\section{External Package}
\label{api:module-experimentdb.external}\label{api:external-package}\index{experimentdb.external (module)}

\subsection{Models}
\label{api:id11}\phantomsection\label{api:module-experimentdb.external.models}\index{experimentdb.external.models (module)}
This package contains the model information for the external app.


It defines the structure and behavior of the following models:
- Contact
- Vendor
- Reference

\index{Contact (class in experimentdb.external.models)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Contact}\pysiglinewithargsret{\strong{class }\code{experimentdb.external.models.}\bfcode{Contact}}{\emph{*args}, \emph{**kwargs}}{}
This model defines a contact.


This is intended to be a person who is involved in the research program, and may be but it not necessarily a database user.
The required fields are first\_name and last\_name.

\index{Contact.DoesNotExist}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Contact.DoesNotExist}\pysigline{\strong{exception }\bfcode{DoesNotExist}}{}
\end{fulllineitems}

\index{Contact.MultipleObjectsReturned}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Contact.MultipleObjectsReturned}\pysigline{\strong{exception }\code{Contact.}\bfcode{MultipleObjectsReturned}}{}
\end{fulllineitems}

\index{antibody\_researcher (experimentdb.external.models.Contact attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Contact.antibody_researcher}\pysigline{\code{Contact.}\bfcode{antibody\_researcher}}{}
\end{fulllineitems}

\index{cell\_researcher (experimentdb.external.models.Contact attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Contact.cell_researcher}\pysigline{\code{Contact.}\bfcode{cell\_researcher}}{}
\end{fulllineitems}

\index{chemical\_researcher (experimentdb.external.models.Contact attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Contact.chemical_researcher}\pysigline{\code{Contact.}\bfcode{chemical\_researcher}}{}
\end{fulllineitems}

\index{cloning\_set (experimentdb.external.models.Contact attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Contact.cloning_set}\pysigline{\code{Contact.}\bfcode{cloning\_set}}{}
\end{fulllineitems}

\index{construct\_researcher (experimentdb.external.models.Contact attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Contact.construct_researcher}\pysigline{\code{Contact.}\bfcode{construct\_researcher}}{}
\end{fulllineitems}

\index{experiment\_set (experimentdb.external.models.Contact attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Contact.experiment_set}\pysigline{\code{Contact.}\bfcode{experiment\_set}}{}
\end{fulllineitems}

\index{get\_absolute\_url() (experimentdb.external.models.Contact method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Contact.get_absolute_url}\pysiglinewithargsret{\code{Contact.}\bfcode{get\_absolute\_url}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{laboratory\_set (experimentdb.external.models.Contact attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Contact.laboratory_set}\pysigline{\code{Contact.}\bfcode{laboratory\_set}}{}
\end{fulllineitems}

\index{mutagenesis\_set (experimentdb.external.models.Contact attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Contact.mutagenesis_set}\pysigline{\code{Contact.}\bfcode{mutagenesis\_set}}{}
\end{fulllineitems}

\index{primer\_researcher (experimentdb.external.models.Contact attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Contact.primer_researcher}\pysigline{\code{Contact.}\bfcode{primer\_researcher}}{}
\end{fulllineitems}

\index{project\_set (experimentdb.external.models.Contact attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Contact.project_set}\pysigline{\code{Contact.}\bfcode{project\_set}}{}
\end{fulllineitems}

\index{reference\_set (experimentdb.external.models.Contact attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Contact.reference_set}\pysigline{\code{Contact.}\bfcode{reference\_set}}{}
\end{fulllineitems}

\index{save() (experimentdb.external.models.Contact method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Contact.save}\pysiglinewithargsret{\code{Contact.}\bfcode{save}}{}{}
The save is over-ridden to slugify the contact field into a slugfield named contactID.


\end{fulllineitems}

\index{strain\_researcher (experimentdb.external.models.Contact attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Contact.strain_researcher}\pysigline{\code{Contact.}\bfcode{strain\_researcher}}{}
\end{fulllineitems}

\index{subproject\_set (experimentdb.external.models.Contact attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Contact.subproject_set}\pysigline{\code{Contact.}\bfcode{subproject\_set}}{}
\end{fulllineitems}

\index{user (experimentdb.external.models.Contact attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Contact.user}\pysigline{\code{Contact.}\bfcode{user}}{}
\end{fulllineitems}


\end{fulllineitems}

\index{Reference (class in experimentdb.external.models)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Reference}\pysiglinewithargsret{\strong{class }\code{experimentdb.external.models.}\bfcode{Reference}}{\emph{*args}, \emph{**kwargs}}{}
This model contains objects of the class reference.


It is intended to keep track of specific papers that pertain to protocols, experiments or projects.


The only required field for this model is a title.

\index{Reference.DoesNotExist}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Reference.DoesNotExist}\pysigline{\strong{exception }\bfcode{DoesNotExist}}{}
\end{fulllineitems}

\index{Reference.MultipleObjectsReturned}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Reference.MultipleObjectsReturned}\pysigline{\strong{exception }\code{Reference.}\bfcode{MultipleObjectsReturned}}{}
\end{fulllineitems}

\index{antibody\_set (experimentdb.external.models.Reference attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Reference.antibody_set}\pysigline{\code{Reference.}\bfcode{antibody\_set}}{}
\end{fulllineitems}

\index{cell\_set (experimentdb.external.models.Reference attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Reference.cell_set}\pysigline{\code{Reference.}\bfcode{cell\_set}}{}
\end{fulllineitems}

\index{chemical\_set (experimentdb.external.models.Reference attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Reference.chemical_set}\pysigline{\code{Reference.}\bfcode{chemical\_set}}{}
\end{fulllineitems}

\index{construct\_set (experimentdb.external.models.Reference attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Reference.construct_set}\pysigline{\code{Reference.}\bfcode{construct\_set}}{}
\end{fulllineitems}

\index{get\_absolute\_url() (experimentdb.external.models.Reference method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Reference.get_absolute_url}\pysiglinewithargsret{\code{Reference.}\bfcode{get\_absolute\_url}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{primer\_set (experimentdb.external.models.Reference attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Reference.primer_set}\pysigline{\code{Reference.}\bfcode{primer\_set}}{}
\end{fulllineitems}

\index{project\_set (experimentdb.external.models.Reference attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Reference.project_set}\pysigline{\code{Reference.}\bfcode{project\_set}}{}
\end{fulllineitems}

\index{researchers (experimentdb.external.models.Reference attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Reference.researchers}\pysigline{\code{Reference.}\bfcode{researchers}}{}
\end{fulllineitems}

\index{strain\_set (experimentdb.external.models.Reference attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Reference.strain_set}\pysigline{\code{Reference.}\bfcode{strain\_set}}{}
\end{fulllineitems}

\index{subproject\_set (experimentdb.external.models.Reference attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Reference.subproject_set}\pysigline{\code{Reference.}\bfcode{subproject\_set}}{}
\end{fulllineitems}


\end{fulllineitems}

\index{Vendor (class in experimentdb.external.models)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Vendor}\pysiglinewithargsret{\strong{class }\code{experimentdb.external.models.}\bfcode{Vendor}}{\emph{*args}, \emph{**kwargs}}{}
This model contains objects of the class vendor.


It is intended to be used to indicate companies from which reagents are obtained.
The only required field is company.

\index{Vendor.DoesNotExist}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Vendor.DoesNotExist}\pysigline{\strong{exception }\bfcode{DoesNotExist}}{}
\end{fulllineitems}

\index{Vendor.MultipleObjectsReturned}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Vendor.MultipleObjectsReturned}\pysigline{\strong{exception }\code{Vendor.}\bfcode{MultipleObjectsReturned}}{}
\end{fulllineitems}

\index{antibody\_vendor (experimentdb.external.models.Vendor attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Vendor.antibody_vendor}\pysigline{\code{Vendor.}\bfcode{antibody\_vendor}}{}
\end{fulllineitems}

\index{cell\_vendor (experimentdb.external.models.Vendor attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Vendor.cell_vendor}\pysigline{\code{Vendor.}\bfcode{cell\_vendor}}{}
\end{fulllineitems}

\index{chemical\_vendor (experimentdb.external.models.Vendor attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Vendor.chemical_vendor}\pysigline{\code{Vendor.}\bfcode{chemical\_vendor}}{}
\end{fulllineitems}

\index{construct\_vendor (experimentdb.external.models.Vendor attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Vendor.construct_vendor}\pysigline{\code{Vendor.}\bfcode{construct\_vendor}}{}
\end{fulllineitems}

\index{get\_absolute\_url() (experimentdb.external.models.Vendor method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Vendor.get_absolute_url}\pysiglinewithargsret{\code{Vendor.}\bfcode{get\_absolute\_url}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{primer\_vendor (experimentdb.external.models.Vendor attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Vendor.primer_vendor}\pysigline{\code{Vendor.}\bfcode{primer\_vendor}}{}
\end{fulllineitems}

\index{save() (experimentdb.external.models.Vendor method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Vendor.save}\pysiglinewithargsret{\code{Vendor.}\bfcode{save}}{}{}
The save is over-ridden to slugify the contact field into a slugfield named contactID.


\end{fulllineitems}

\index{strain\_vendor (experimentdb.external.models.Vendor attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.models.Vendor.strain_vendor}\pysigline{\code{Vendor.}\bfcode{strain\_vendor}}{}
\end{fulllineitems}


\end{fulllineitems}



\subsection{Views}
\label{api:id12}\phantomsection\label{api:module-experimentdb.external.views}\index{experimentdb.external.views (module)}

\subsection{URLconfs}
\label{api:id13}\phantomsection\label{api:module-experimentdb.external.urls}\index{experimentdb.external.urls (module)}
This folder contains the urlconf redirections for the external app.


There is separate files for vendor, contact and reference urls.



\subsection{Tests}
\label{api:id14}\phantomsection\label{api:module-experimentdb.external.tests}\index{experimentdb.external.tests (module)}
This package defines the tests for the external app.


It contains model tests for the models:
- Vendor
- Reference
- Contact


There are currently no views associated with these models.

\index{ContactModelTests (class in experimentdb.external.tests)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.tests.ContactModelTests}\pysiglinewithargsret{\strong{class }\code{experimentdb.external.tests.}\bfcode{ContactModelTests}}{\emph{methodName='runTest'}}{}
Tests the model attributes of Contact objects contained in the reagents app.

\index{setUp() (experimentdb.external.tests.ContactModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.tests.ContactModelTests.setUp}\pysiglinewithargsret{\bfcode{setUp}}{}{}
Instantiate the test client.


\end{fulllineitems}

\index{tearDown() (experimentdb.external.tests.ContactModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.tests.ContactModelTests.tearDown}\pysiglinewithargsret{\bfcode{tearDown}}{}{}
Depopulate created model instances from test database.


\end{fulllineitems}

\index{test\_contact\_absolute\_url() (experimentdb.external.tests.ContactModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.tests.ContactModelTests.test_contact_absolute_url}\pysiglinewithargsret{\bfcode{test\_contact\_absolute\_url}}{}{}
\end{fulllineitems}

\index{test\_contact\_slugify() (experimentdb.external.tests.ContactModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.tests.ContactModelTests.test_contact_slugify}\pysiglinewithargsret{\bfcode{test\_contact\_slugify}}{}{}
\end{fulllineitems}

\index{test\_create\_contact\_minimal() (experimentdb.external.tests.ContactModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.tests.ContactModelTests.test_create_contact_minimal}\pysiglinewithargsret{\bfcode{test\_create\_contact\_minimal}}{}{}
This is a test for creating a new primer object, with only the minimum fields being entered


\end{fulllineitems}


\end{fulllineitems}

\index{ReferenceModelTests (class in experimentdb.external.tests)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.tests.ReferenceModelTests}\pysiglinewithargsret{\strong{class }\code{experimentdb.external.tests.}\bfcode{ReferenceModelTests}}{\emph{methodName='runTest'}}{}
Tests the model attributes of Reference objects contained in the reagents app.

\index{setUp() (experimentdb.external.tests.ReferenceModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.tests.ReferenceModelTests.setUp}\pysiglinewithargsret{\bfcode{setUp}}{}{}
Instantiate the test client.


\end{fulllineitems}

\index{tearDown() (experimentdb.external.tests.ReferenceModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.tests.ReferenceModelTests.tearDown}\pysiglinewithargsret{\bfcode{tearDown}}{}{}
Depopulate created model instances from test database.


\end{fulllineitems}

\index{test\_create\_reference\_minimal() (experimentdb.external.tests.ReferenceModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.tests.ReferenceModelTests.test_create_reference_minimal}\pysiglinewithargsret{\bfcode{test\_create\_reference\_minimal}}{}{}
This is a test for creating a new primer object, with only the minimum fields being entered


\end{fulllineitems}

\index{test\_reference\_absolute\_url() (experimentdb.external.tests.ReferenceModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.tests.ReferenceModelTests.test_reference_absolute_url}\pysiglinewithargsret{\bfcode{test\_reference\_absolute\_url}}{}{}
\end{fulllineitems}


\end{fulllineitems}

\index{VendorModelTests (class in experimentdb.external.tests)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.tests.VendorModelTests}\pysiglinewithargsret{\strong{class }\code{experimentdb.external.tests.}\bfcode{VendorModelTests}}{\emph{methodName='runTest'}}{}
Tests the model attributes of Vendor objects contained in the reagents app.

\index{setUp() (experimentdb.external.tests.VendorModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.tests.VendorModelTests.setUp}\pysiglinewithargsret{\bfcode{setUp}}{}{}
Instantiate the test client.


\end{fulllineitems}

\index{tearDown() (experimentdb.external.tests.VendorModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.tests.VendorModelTests.tearDown}\pysiglinewithargsret{\bfcode{tearDown}}{}{}
Depopulate created model instances from test database.


\end{fulllineitems}

\index{test\_create\_vendor\_minimal() (experimentdb.external.tests.VendorModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.tests.VendorModelTests.test_create_vendor_minimal}\pysiglinewithargsret{\bfcode{test\_create\_vendor\_minimal}}{}{}
This is a test for creating a new primer object, with only the minimum fields being entered


\end{fulllineitems}

\index{test\_vendor\_absolute\_url() (experimentdb.external.tests.VendorModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.external.tests.VendorModelTests.test_vendor_absolute_url}\pysiglinewithargsret{\bfcode{test\_vendor\_absolute\_url}}{}{}
\end{fulllineitems}


\end{fulllineitems}



\subsection{Fixtures}
\label{api:id15}\phantomsection\label{api:module-experimentdb.external.fixtures}\index{experimentdb.external.fixtures (module)}
This folder contains test fixtures for the external app.


Currently there is one file with a sample object for each of:
- Vendor
- Contact
- Reference


in the file test\_external.json



\section{Proteins Package}
\label{api:module-experimentdb.proteins}\label{api:proteins-package}\index{experimentdb.proteins (module)}

\subsection{Models}
\label{api:id16}\phantomsection\label{api:module-experimentdb.proteins.models}\index{experimentdb.proteins.models (module)}\index{Protein (class in experimentdb.proteins.models)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.models.Protein}\pysiglinewithargsret{\strong{class }\code{experimentdb.proteins.models.}\bfcode{Protein}}{\emph{*args}, \emph{**kwargs}}{}
Protein(id, name)

\index{Protein.DoesNotExist}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.models.Protein.DoesNotExist}\pysigline{\strong{exception }\bfcode{DoesNotExist}}{}
\end{fulllineitems}

\index{Protein.MultipleObjectsReturned}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.models.Protein.MultipleObjectsReturned}\pysigline{\strong{exception }\code{Protein.}\bfcode{MultipleObjectsReturned}}{}
\end{fulllineitems}

\index{antibody\_set (experimentdb.proteins.models.Protein attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.models.Protein.antibody_set}\pysigline{\code{Protein.}\bfcode{antibody\_set}}{}
\end{fulllineitems}

\index{cell\_set (experimentdb.proteins.models.Protein attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.models.Protein.cell_set}\pysigline{\code{Protein.}\bfcode{cell\_set}}{}
\end{fulllineitems}

\index{chemical\_set (experimentdb.proteins.models.Protein attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.models.Protein.chemical_set}\pysigline{\code{Protein.}\bfcode{chemical\_set}}{}
\end{fulllineitems}

\index{construct\_set (experimentdb.proteins.models.Protein attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.models.Protein.construct_set}\pysigline{\code{Protein.}\bfcode{construct\_set}}{}
\end{fulllineitems}

\index{experiment\_set (experimentdb.proteins.models.Protein attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.models.Protein.experiment_set}\pysigline{\code{Protein.}\bfcode{experiment\_set}}{}
\end{fulllineitems}

\index{get\_absolute\_url() (experimentdb.proteins.models.Protein method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.models.Protein.get_absolute_url}\pysiglinewithargsret{\code{Protein.}\bfcode{get\_absolute\_url}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{primer\_set (experimentdb.proteins.models.Protein attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.models.Protein.primer_set}\pysigline{\code{Protein.}\bfcode{primer\_set}}{}
\end{fulllineitems}

\index{protein\_family (experimentdb.proteins.models.Protein attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.models.Protein.protein_family}\pysigline{\code{Protein.}\bfcode{protein\_family}}{}
\end{fulllineitems}

\index{proteindetail\_set (experimentdb.proteins.models.Protein attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.models.Protein.proteindetail_set}\pysigline{\code{Protein.}\bfcode{proteindetail\_set}}{}
\end{fulllineitems}

\index{strain\_set (experimentdb.proteins.models.Protein attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.models.Protein.strain_set}\pysigline{\code{Protein.}\bfcode{strain\_set}}{}
\end{fulllineitems}


\end{fulllineitems}

\index{ProteinDetail (class in experimentdb.proteins.models)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.models.ProteinDetail}\pysiglinewithargsret{\strong{class }\code{experimentdb.proteins.models.}\bfcode{ProteinDetail}}{\emph{*args}, \emph{**kwargs}}{}
ProteinDetail(id, name, protein\_id, gene, species\_id, geneID, RefSeqProtein, RefSeqProtein\_gi, RefSeqNucleotide, RefSeqNucleotide\_gi, WormBaseID, FlyBaseID, SGD\_ID, public, published)

\index{ProteinDetail.DoesNotExist}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.models.ProteinDetail.DoesNotExist}\pysigline{\strong{exception }\bfcode{DoesNotExist}}{}
\end{fulllineitems}

\index{ProteinDetail.MultipleObjectsReturned}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.models.ProteinDetail.MultipleObjectsReturned}\pysigline{\strong{exception }\code{ProteinDetail.}\bfcode{MultipleObjectsReturned}}{}
\end{fulllineitems}

\index{protein (experimentdb.proteins.models.ProteinDetail attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.models.ProteinDetail.protein}\pysigline{\code{ProteinDetail.}\bfcode{protein}}{}
\end{fulllineitems}

\index{species (experimentdb.proteins.models.ProteinDetail attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.models.ProteinDetail.species}\pysigline{\code{ProteinDetail.}\bfcode{species}}{}
\end{fulllineitems}


\end{fulllineitems}

\index{ProteinFamily (class in experimentdb.proteins.models)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.models.ProteinFamily}\pysiglinewithargsret{\strong{class }\code{experimentdb.proteins.models.}\bfcode{ProteinFamily}}{\emph{*args}, \emph{**kwargs}}{}
ProteinFamily(id, name, notes)

\index{ProteinFamily.DoesNotExist}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.models.ProteinFamily.DoesNotExist}\pysigline{\strong{exception }\bfcode{DoesNotExist}}{}
\end{fulllineitems}

\index{ProteinFamily.MultipleObjectsReturned}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.models.ProteinFamily.MultipleObjectsReturned}\pysigline{\strong{exception }\code{ProteinFamily.}\bfcode{MultipleObjectsReturned}}{}
\end{fulllineitems}

\index{get\_absolute\_url() (experimentdb.proteins.models.ProteinFamily method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.models.ProteinFamily.get_absolute_url}\pysiglinewithargsret{\code{ProteinFamily.}\bfcode{get\_absolute\_url}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{protein\_set (experimentdb.proteins.models.ProteinFamily attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.models.ProteinFamily.protein_set}\pysigline{\code{ProteinFamily.}\bfcode{protein\_set}}{}
\end{fulllineitems}


\end{fulllineitems}

\index{Species (class in experimentdb.proteins.models)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.models.Species}\pysiglinewithargsret{\strong{class }\code{experimentdb.proteins.models.}\bfcode{Species}}{\emph{*args}, \emph{**kwargs}}{}
Species(id, common\_name, scientific\_name, taxonomy\_id)

\index{Species.DoesNotExist}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.models.Species.DoesNotExist}\pysigline{\strong{exception }\bfcode{DoesNotExist}}{}
\end{fulllineitems}

\index{Species.MultipleObjectsReturned}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.models.Species.MultipleObjectsReturned}\pysigline{\strong{exception }\code{Species.}\bfcode{MultipleObjectsReturned}}{}
\end{fulllineitems}

\index{proteindetail\_set (experimentdb.proteins.models.Species attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.models.Species.proteindetail_set}\pysigline{\code{Species.}\bfcode{proteindetail\_set}}{}
\end{fulllineitems}


\end{fulllineitems}



\subsection{Views}
\label{api:id17}\phantomsection\label{api:module-experimentdb.proteins.views}\index{experimentdb.proteins.views (module)}\index{detail() (in module experimentdb.proteins.views)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.views.detail}\pysiglinewithargsret{\code{experimentdb.proteins.views.}\bfcode{detail}}{\emph{request}, \emph{*args}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{index() (in module experimentdb.proteins.views)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.views.index}\pysiglinewithargsret{\code{experimentdb.proteins.views.}\bfcode{index}}{\emph{request}, \emph{*args}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{protein\_isoform\_detail() (in module experimentdb.proteins.views)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.views.protein_isoform_detail}\pysiglinewithargsret{\code{experimentdb.proteins.views.}\bfcode{protein\_isoform\_detail}}{\emph{request}, \emph{*args}, \emph{**kwargs}}{}
fetch and parse a genbank protein record


\end{fulllineitems}



\subsection{Lookups}
\label{api:id18}\phantomsection\label{api:module-experimentdb.proteins.lookups}\index{experimentdb.proteins.lookups (module)}
This is a configuration file for the ajax lookups for the proteins app.


See \href{http://code.google.com/p/django-ajax-selects/}{http://code.google.com/p/django-ajax-selects/} for information about configuring the ajax lookups.

\index{ProteinLookup (class in experimentdb.proteins.lookups)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.lookups.ProteinLookup}\pysigline{\strong{class }\code{experimentdb.proteins.lookups.}\bfcode{ProteinLookup}}{}
This is the generic lookup for antibodies.


It is to be used for all protein requests and directs to the `protein' channel.

\index{format\_item() (experimentdb.proteins.lookups.ProteinLookup method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.lookups.ProteinLookup.format_item}\pysiglinewithargsret{\bfcode{format\_item}}{\emph{protein}}{}
the display of a currently selected object in the area below the search box. html is OK


\end{fulllineitems}

\index{format\_result() (experimentdb.proteins.lookups.ProteinLookup method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.lookups.ProteinLookup.format_result}\pysiglinewithargsret{\bfcode{format\_result}}{\emph{protein}}{}
This controls the display of the dropdown menu.


This is set to show the unicode name of the protein.


\end{fulllineitems}

\index{get\_objects() (experimentdb.proteins.lookups.ProteinLookup method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.lookups.ProteinLookup.get_objects}\pysiglinewithargsret{\bfcode{get\_objects}}{\emph{ids}}{}
given a list of ids, return the objects ordered as you would like them on the admin page.
this is for displaying the currently selected items (in the case of a ManyToMany field)


\end{fulllineitems}

\index{get\_query() (experimentdb.proteins.lookups.ProteinLookup method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.lookups.ProteinLookup.get_query}\pysiglinewithargsret{\bfcode{get\_query}}{\emph{q}, \emph{request}}{}
This sets up the query for the lookup.


The lookup searches the name of the protein.


\end{fulllineitems}


\end{fulllineitems}



\subsection{URLconfs}
\label{api:id19}\phantomsection\label{api:module-experimentdb.proteins.urls}\index{experimentdb.proteins.urls (module)}\index{restricted\_change\_protein() (in module experimentdb.proteins.urls)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.urls.restricted_change_protein}\pysiglinewithargsret{\code{experimentdb.proteins.urls.}\bfcode{restricted\_change\_protein}}{\emph{request}, \emph{*args}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{restricted\_create\_protein() (in module experimentdb.proteins.urls)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.urls.restricted_create_protein}\pysiglinewithargsret{\code{experimentdb.proteins.urls.}\bfcode{restricted\_create\_protein}}{\emph{request}, \emph{*args}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{restricted\_delete\_protein() (in module experimentdb.proteins.urls)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.urls.restricted_delete_protein}\pysiglinewithargsret{\code{experimentdb.proteins.urls.}\bfcode{restricted\_delete\_protein}}{\emph{request}, \emph{*args}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{restricted\_detail() (in module experimentdb.proteins.urls)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.urls.restricted_detail}\pysiglinewithargsret{\code{experimentdb.proteins.urls.}\bfcode{restricted\_detail}}{\emph{request}, \emph{*args}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{restricted\_object\_list() (in module experimentdb.proteins.urls)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.proteins.urls.restricted_object_list}\pysiglinewithargsret{\code{experimentdb.proteins.urls.}\bfcode{restricted\_object\_list}}{\emph{request}, \emph{*args}, \emph{**kwargs}}{}
\end{fulllineitems}



\subsection{Tests}
\label{api:id20}

\subsection{Fixtures}
\label{api:id21}\phantomsection\label{api:module-experimentdb.proteins.fixtures}\index{experimentdb.proteins.fixtures (module)}
This folder contains data fixtures for the proteins app.


Currently there is one text fixture for Protein objects named test\_protein.json.



\section{Reagents Package}
\label{api:module-experimentdb.reagents}\label{api:reagents-package}\index{experimentdb.reagents (module)}

\subsection{Models}
\label{api:id22}\phantomsection\label{api:module-experimentdb.reagents.models}\index{experimentdb.reagents.models (module)}
This package describes the models in the reagents app.


The models are ReagentInfo, which is an abstract superclass of:
- Primer
- Cell
- Antibody
- Strain
- Chemical
- Construct


The ReagentInfo class provides generic fields to all the models, while each subclass provides extra specific fields.
This package also contains a Selection model, to be used for antibiotic selections, and a specied model, to be used to indicate various species.

\index{Antibody (class in experimentdb.reagents.models)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Antibody}\pysiglinewithargsret{\strong{class }\code{experimentdb.reagents.models.}\bfcode{Antibody}}{\emph{*args}, \emph{**kwargs}}{}
This model describes antibodies.


The required fields are name and source\_species.
This model is a subclass of ReagentInfo.

\index{Antibody.DoesNotExist}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Antibody.DoesNotExist}\pysigline{\strong{exception }\bfcode{DoesNotExist}}{}
\end{fulllineitems}

\index{Antibody.MultipleObjectsReturned}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Antibody.MultipleObjectsReturned}\pysigline{\strong{exception }\code{Antibody.}\bfcode{MultipleObjectsReturned}}{}
\end{fulllineitems}

\index{experiment\_set (experimentdb.reagents.models.Antibody attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Antibody.experiment_set}\pysigline{\code{Antibody.}\bfcode{experiment\_set}}{}
\end{fulllineitems}

\index{get\_absolute\_url() (experimentdb.reagents.models.Antibody method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Antibody.get_absolute_url}\pysiglinewithargsret{\code{Antibody.}\bfcode{get\_absolute\_url}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{get\_location\_display() (experimentdb.reagents.models.Antibody method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Antibody.get_location_display}\pysiglinewithargsret{\code{Antibody.}\bfcode{get\_location\_display}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{get\_source\_species\_display() (experimentdb.reagents.models.Antibody method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Antibody.get_source_species_display}\pysiglinewithargsret{\code{Antibody.}\bfcode{get\_source\_species\_display}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{protein (experimentdb.reagents.models.Antibody attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Antibody.protein}\pysigline{\code{Antibody.}\bfcode{protein}}{}
\end{fulllineitems}

\index{reference (experimentdb.reagents.models.Antibody attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Antibody.reference}\pysigline{\code{Antibody.}\bfcode{reference}}{}
\end{fulllineitems}

\index{researcher (experimentdb.reagents.models.Antibody attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Antibody.researcher}\pysigline{\code{Antibody.}\bfcode{researcher}}{}
\end{fulllineitems}

\index{save() (experimentdb.reagents.models.Antibody method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Antibody.save}\pysiglinewithargsret{\code{Antibody.}\bfcode{save}}{}{}
The save is over-ridden to slugify the name field into a slugfield.


\end{fulllineitems}

\index{species (experimentdb.reagents.models.Antibody attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Antibody.species}\pysigline{\code{Antibody.}\bfcode{species}}{}
\end{fulllineitems}

\index{vendor (experimentdb.reagents.models.Antibody attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Antibody.vendor}\pysigline{\code{Antibody.}\bfcode{vendor}}{}
\end{fulllineitems}


\end{fulllineitems}

\index{Cell (class in experimentdb.reagents.models)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Cell}\pysiglinewithargsret{\strong{class }\code{experimentdb.reagents.models.}\bfcode{Cell}}{\emph{*args}, \emph{**kwargs}}{}
This model describes objects of the class Cell.


This model is intended to be used to store information about mammalian cell lines.
The only required field is name.
This model is a subclass of ReagentInfo.

\index{Cell.DoesNotExist}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Cell.DoesNotExist}\pysigline{\strong{exception }\bfcode{DoesNotExist}}{}
\end{fulllineitems}

\index{Cell.MultipleObjectsReturned}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Cell.MultipleObjectsReturned}\pysigline{\strong{exception }\code{Cell.}\bfcode{MultipleObjectsReturned}}{}
\end{fulllineitems}

\index{cell\_line\_species (experimentdb.reagents.models.Cell attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Cell.cell_line_species}\pysigline{\code{Cell.}\bfcode{cell\_line\_species}}{}
\end{fulllineitems}

\index{experiment\_set (experimentdb.reagents.models.Cell attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Cell.experiment_set}\pysigline{\code{Cell.}\bfcode{experiment\_set}}{}
\end{fulllineitems}

\index{get\_absolute\_url() (experimentdb.reagents.models.Cell method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Cell.get_absolute_url}\pysiglinewithargsret{\code{Cell.}\bfcode{get\_absolute\_url}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{get\_location\_display() (experimentdb.reagents.models.Cell method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Cell.get_location_display}\pysiglinewithargsret{\code{Cell.}\bfcode{get\_location\_display}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{get\_species\_display() (experimentdb.reagents.models.Cell method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Cell.get_species_display}\pysiglinewithargsret{\code{Cell.}\bfcode{get\_species\_display}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{protein (experimentdb.reagents.models.Cell attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Cell.protein}\pysigline{\code{Cell.}\bfcode{protein}}{}
\end{fulllineitems}

\index{reference (experimentdb.reagents.models.Cell attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Cell.reference}\pysigline{\code{Cell.}\bfcode{reference}}{}
\end{fulllineitems}

\index{researcher (experimentdb.reagents.models.Cell attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Cell.researcher}\pysigline{\code{Cell.}\bfcode{researcher}}{}
\end{fulllineitems}

\index{save() (experimentdb.reagents.models.Cell method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Cell.save}\pysiglinewithargsret{\code{Cell.}\bfcode{save}}{}{}
The save is over-ridden to slugify the name field into a slugfield.


\end{fulllineitems}

\index{vendor (experimentdb.reagents.models.Cell attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Cell.vendor}\pysigline{\code{Cell.}\bfcode{vendor}}{}
\end{fulllineitems}


\end{fulllineitems}

\index{Chemical (class in experimentdb.reagents.models)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Chemical}\pysiglinewithargsret{\strong{class }\code{experimentdb.reagents.models.}\bfcode{Chemical}}{\emph{*args}, \emph{**kwargs}}{}
This model describes objects of the class Chemical.


It is intended to describe chemicals used in experiments.
The only required field is name.
This model is a subclass of ReagentInfo.

\index{Chemical.DoesNotExist}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Chemical.DoesNotExist}\pysigline{\strong{exception }\bfcode{DoesNotExist}}{}
\end{fulllineitems}

\index{Chemical.MultipleObjectsReturned}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Chemical.MultipleObjectsReturned}\pysigline{\strong{exception }\code{Chemical.}\bfcode{MultipleObjectsReturned}}{}
\end{fulllineitems}

\index{experiment\_set (experimentdb.reagents.models.Chemical attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Chemical.experiment_set}\pysigline{\code{Chemical.}\bfcode{experiment\_set}}{}
\end{fulllineitems}

\index{get\_absolute\_url() (experimentdb.reagents.models.Chemical method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Chemical.get_absolute_url}\pysiglinewithargsret{\code{Chemical.}\bfcode{get\_absolute\_url}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{get\_location\_display() (experimentdb.reagents.models.Chemical method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Chemical.get_location_display}\pysiglinewithargsret{\code{Chemical.}\bfcode{get\_location\_display}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{protein (experimentdb.reagents.models.Chemical attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Chemical.protein}\pysigline{\code{Chemical.}\bfcode{protein}}{}
\end{fulllineitems}

\index{reference (experimentdb.reagents.models.Chemical attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Chemical.reference}\pysigline{\code{Chemical.}\bfcode{reference}}{}
\end{fulllineitems}

\index{researcher (experimentdb.reagents.models.Chemical attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Chemical.researcher}\pysigline{\code{Chemical.}\bfcode{researcher}}{}
\end{fulllineitems}

\index{save() (experimentdb.reagents.models.Chemical method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Chemical.save}\pysiglinewithargsret{\code{Chemical.}\bfcode{save}}{}{}
The save is over-ridden to slugify the name field into a slugfield.


\end{fulllineitems}

\index{vendor (experimentdb.reagents.models.Chemical attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Chemical.vendor}\pysigline{\code{Chemical.}\bfcode{vendor}}{}
\end{fulllineitems}


\end{fulllineitems}

\index{Construct (class in experimentdb.reagents.models)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Construct}\pysiglinewithargsret{\strong{class }\code{experimentdb.reagents.models.}\bfcode{Construct}}{\emph{*args}, \emph{**kwargs}}{}
This model describes recombinant DNA objects.


The only required field is name.
It is a subclass of ReagentInfo.

\index{Construct.DoesNotExist}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Construct.DoesNotExist}\pysigline{\strong{exception }\bfcode{DoesNotExist}}{}
\end{fulllineitems}

\index{Construct.MultipleObjectsReturned}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Construct.MultipleObjectsReturned}\pysigline{\strong{exception }\code{Construct.}\bfcode{MultipleObjectsReturned}}{}
\end{fulllineitems}

\index{constructshipment\_set (experimentdb.reagents.models.Construct attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Construct.constructshipment_set}\pysigline{\code{Construct.}\bfcode{constructshipment\_set}}{}
\end{fulllineitems}

\index{experiment\_set (experimentdb.reagents.models.Construct attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Construct.experiment_set}\pysigline{\code{Construct.}\bfcode{experiment\_set}}{}
\end{fulllineitems}

\index{final\_clone (experimentdb.reagents.models.Construct attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Construct.final_clone}\pysigline{\code{Construct.}\bfcode{final\_clone}}{}
\end{fulllineitems}

\index{get\_absolute\_url() (experimentdb.reagents.models.Construct method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Construct.get_absolute_url}\pysiglinewithargsret{\code{Construct.}\bfcode{get\_absolute\_url}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{get\_location\_display() (experimentdb.reagents.models.Construct method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Construct.get_location_display}\pysiglinewithargsret{\code{Construct.}\bfcode{get\_location\_display}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{mutant (experimentdb.reagents.models.Construct attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Construct.mutant}\pysigline{\code{Construct.}\bfcode{mutant}}{}
\end{fulllineitems}

\index{protein (experimentdb.reagents.models.Construct attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Construct.protein}\pysigline{\code{Construct.}\bfcode{protein}}{}
\end{fulllineitems}

\index{recipient\_vector (experimentdb.reagents.models.Construct attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Construct.recipient_vector}\pysigline{\code{Construct.}\bfcode{recipient\_vector}}{}
\end{fulllineitems}

\index{reference (experimentdb.reagents.models.Construct attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Construct.reference}\pysigline{\code{Construct.}\bfcode{reference}}{}
\end{fulllineitems}

\index{researcher (experimentdb.reagents.models.Construct attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Construct.researcher}\pysigline{\code{Construct.}\bfcode{researcher}}{}
\end{fulllineitems}

\index{save() (experimentdb.reagents.models.Construct method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Construct.save}\pysiglinewithargsret{\code{Construct.}\bfcode{save}}{}{}
The save is over-ridden to slugify the name field into a slugfield.


\end{fulllineitems}

\index{selection (experimentdb.reagents.models.Construct attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Construct.selection}\pysigline{\code{Construct.}\bfcode{selection}}{}
\end{fulllineitems}

\index{sequencing\_set (experimentdb.reagents.models.Construct attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Construct.sequencing_set}\pysigline{\code{Construct.}\bfcode{sequencing\_set}}{}
\end{fulllineitems}

\index{strain\_set (experimentdb.reagents.models.Construct attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Construct.strain_set}\pysigline{\code{Construct.}\bfcode{strain\_set}}{}
\end{fulllineitems}

\index{template (experimentdb.reagents.models.Construct attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Construct.template}\pysigline{\code{Construct.}\bfcode{template}}{}
\end{fulllineitems}

\index{vendor (experimentdb.reagents.models.Construct attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Construct.vendor}\pysigline{\code{Construct.}\bfcode{vendor}}{}
\end{fulllineitems}


\end{fulllineitems}

\index{Primer (class in experimentdb.reagents.models)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Primer}\pysiglinewithargsret{\strong{class }\code{experimentdb.reagents.models.}\bfcode{Primer}}{\emph{*args}, \emph{**kwargs}}{}
Model describing primer objects.


These objects can be of any short nucleotide type including primers, siRNA oligos or morpholinos.  
The required fields are the name and the type.  
The nonrequired fields include the sequence, the protein, the ordering date and all generic reagent info fields.
This is a subclass of the ReagentInfo abstract base class.

\index{3\_Primer (experimentdb.reagents.models.Primer attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Primer.3_Primer}\pysigline{\bfcode{3\_Primer}}{}
\end{fulllineitems}

\index{5\_Primer (experimentdb.reagents.models.Primer attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Primer.5_Primer}\pysigline{\bfcode{5\_Primer}}{}
\end{fulllineitems}

\index{Primer.DoesNotExist}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Primer.DoesNotExist}\pysigline{\strong{exception }\bfcode{DoesNotExist}}{}
\end{fulllineitems}

\index{Primer.MultipleObjectsReturned}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Primer.MultipleObjectsReturned}\pysigline{\strong{exception }\code{Primer.}\bfcode{MultipleObjectsReturned}}{}
\end{fulllineitems}

\index{antisense\_primer (experimentdb.reagents.models.Primer attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Primer.antisense_primer}\pysigline{\code{Primer.}\bfcode{antisense\_primer}}{}
\end{fulllineitems}

\index{experiment\_set (experimentdb.reagents.models.Primer attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Primer.experiment_set}\pysigline{\code{Primer.}\bfcode{experiment\_set}}{}
\end{fulllineitems}

\index{get\_absolute\_url() (experimentdb.reagents.models.Primer method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Primer.get_absolute_url}\pysiglinewithargsret{\code{Primer.}\bfcode{get\_absolute\_url}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{get\_location\_display() (experimentdb.reagents.models.Primer method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Primer.get_location_display}\pysiglinewithargsret{\code{Primer.}\bfcode{get\_location\_display}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{get\_primer\_type\_display() (experimentdb.reagents.models.Primer method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Primer.get_primer_type_display}\pysiglinewithargsret{\code{Primer.}\bfcode{get\_primer\_type\_display}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{protein (experimentdb.reagents.models.Primer attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Primer.protein}\pysigline{\code{Primer.}\bfcode{protein}}{}
\end{fulllineitems}

\index{reference (experimentdb.reagents.models.Primer attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Primer.reference}\pysigline{\code{Primer.}\bfcode{reference}}{}
\end{fulllineitems}

\index{researcher (experimentdb.reagents.models.Primer attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Primer.researcher}\pysigline{\code{Primer.}\bfcode{researcher}}{}
\end{fulllineitems}

\index{save() (experimentdb.reagents.models.Primer method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Primer.save}\pysiglinewithargsret{\code{Primer.}\bfcode{save}}{}{}
The save is over-ridden to slugify the name field into a slugfield.


\end{fulllineitems}

\index{sense\_primer (experimentdb.reagents.models.Primer attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Primer.sense_primer}\pysigline{\code{Primer.}\bfcode{sense\_primer}}{}
\end{fulllineitems}

\index{sequencing\_set (experimentdb.reagents.models.Primer attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Primer.sequencing_set}\pysigline{\code{Primer.}\bfcode{sequencing\_set}}{}
\end{fulllineitems}

\index{vendor (experimentdb.reagents.models.Primer attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Primer.vendor}\pysigline{\code{Primer.}\bfcode{vendor}}{}
\end{fulllineitems}


\end{fulllineitems}

\index{ReagentInfo (class in experimentdb.reagents.models)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.ReagentInfo}\pysiglinewithargsret{\strong{class }\code{experimentdb.reagents.models.}\bfcode{ReagentInfo}}{\emph{*args}, \emph{**kwargs}}{}
Abstract base model for all reagents, will not be used in isolation, only as part of other models.


This superclass provides several generic fields, available to all reagents.  The only required field of all reagents is name.
It orders all reagents by name, although this may be over-ridden in the model.  
It also sets sets their \_\_unicode\_\_ representation to be ``name''.

\index{ReagentInfo.Meta (class in experimentdb.reagents.models)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.ReagentInfo.Meta}\pysigline{\strong{class }\bfcode{Meta}}{}
\end{fulllineitems}

\index{get\_location\_display() (experimentdb.reagents.models.ReagentInfo method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.ReagentInfo.get_location_display}\pysiglinewithargsret{\code{ReagentInfo.}\bfcode{get\_location\_display}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{protein (experimentdb.reagents.models.ReagentInfo attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.ReagentInfo.protein}\pysigline{\code{ReagentInfo.}\bfcode{protein}}{}
\end{fulllineitems}

\index{reference (experimentdb.reagents.models.ReagentInfo attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.ReagentInfo.reference}\pysigline{\code{ReagentInfo.}\bfcode{reference}}{}
\end{fulllineitems}

\index{researcher (experimentdb.reagents.models.ReagentInfo attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.ReagentInfo.researcher}\pysigline{\code{ReagentInfo.}\bfcode{researcher}}{}
\end{fulllineitems}

\index{vendor (experimentdb.reagents.models.ReagentInfo attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.ReagentInfo.vendor}\pysigline{\code{ReagentInfo.}\bfcode{vendor}}{}
\end{fulllineitems}


\end{fulllineitems}

\index{Selection (class in experimentdb.reagents.models)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Selection}\pysiglinewithargsret{\strong{class }\code{experimentdb.reagents.models.}\bfcode{Selection}}{\emph{*args}, \emph{**kwargs}}{}
Model for selection conditions of transformants.


This object has one required field, being \textbf{selection}.  An optional comments field is also available.


Initial data upon installation includes resistance to ampicillin or kanamycin.  Other selective markers should be added at /experimentdb/selection/new

\index{Selection.DoesNotExist}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Selection.DoesNotExist}\pysigline{\strong{exception }\bfcode{DoesNotExist}}{}
\end{fulllineitems}

\index{Selection.MultipleObjectsReturned}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Selection.MultipleObjectsReturned}\pysigline{\strong{exception }\code{Selection.}\bfcode{MultipleObjectsReturned}}{}
\end{fulllineitems}

\index{construct\_set (experimentdb.reagents.models.Selection attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Selection.construct_set}\pysigline{\code{Selection.}\bfcode{construct\_set}}{}
\end{fulllineitems}

\index{get\_absolute\_url() (experimentdb.reagents.models.Selection method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Selection.get_absolute_url}\pysiglinewithargsret{\code{Selection.}\bfcode{get\_absolute\_url}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{save() (experimentdb.reagents.models.Selection method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Selection.save}\pysiglinewithargsret{\code{Selection.}\bfcode{save}}{}{}
The save is over-ridden to slugify the selection field into a slugfield.


\end{fulllineitems}

\index{strain\_set (experimentdb.reagents.models.Selection attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Selection.strain_set}\pysigline{\code{Selection.}\bfcode{strain\_set}}{}
\end{fulllineitems}


\end{fulllineitems}

\index{Species (class in experimentdb.reagents.models)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Species}\pysiglinewithargsret{\strong{class }\code{experimentdb.reagents.models.}\bfcode{Species}}{\emph{*args}, \emph{**kwargs}}{}
Model for indicating specific species.


The only required field is common\_name.
This is used with Strain, Cell and Antibody objects.
Currently the species field, with the old choices=SPECIES is present until data can be migrated.  
Upon installation, initial data is provided for rabbit, mouse, human, yeast and goat species.  
More species can be added at /experimentdb/species/new.

\index{Species.DoesNotExist}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Species.DoesNotExist}\pysigline{\strong{exception }\bfcode{DoesNotExist}}{}
\end{fulllineitems}

\index{Species.MultipleObjectsReturned}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Species.MultipleObjectsReturned}\pysigline{\strong{exception }\code{Species.}\bfcode{MultipleObjectsReturned}}{}
\end{fulllineitems}

\index{antibody\_set (experimentdb.reagents.models.Species attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Species.antibody_set}\pysigline{\code{Species.}\bfcode{antibody\_set}}{}
\end{fulllineitems}

\index{cell\_set (experimentdb.reagents.models.Species attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Species.cell_set}\pysigline{\code{Species.}\bfcode{cell\_set}}{}
\end{fulllineitems}

\index{get\_absolute\_url() (experimentdb.reagents.models.Species method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Species.get_absolute_url}\pysiglinewithargsret{\code{Species.}\bfcode{get\_absolute\_url}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{save() (experimentdb.reagents.models.Species method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Species.save}\pysiglinewithargsret{\code{Species.}\bfcode{save}}{}{}
The save is over-ridden to slugify the common\_name field into a slugfield.


\end{fulllineitems}

\index{strain\_set (experimentdb.reagents.models.Species attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Species.strain_set}\pysigline{\code{Species.}\bfcode{strain\_set}}{}
\end{fulllineitems}


\end{fulllineitems}

\index{Strain (class in experimentdb.reagents.models)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Strain}\pysiglinewithargsret{\strong{class }\code{experimentdb.reagents.models.}\bfcode{Strain}}{\emph{*args}, \emph{**kwargs}}{}
Model describing biological strains.


This was devised to organize yeast strains, but can be used for bacteria or other organisms as well.
The only required field is \textbf{name}.
This is a subclass of ReagentInfo abstract class

\index{Strain.DoesNotExist}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Strain.DoesNotExist}\pysigline{\strong{exception }\bfcode{DoesNotExist}}{}
\end{fulllineitems}

\index{Strain.MultipleObjectsReturned}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Strain.MultipleObjectsReturned}\pysigline{\strong{exception }\code{Strain.}\bfcode{MultipleObjectsReturned}}{}
\end{fulllineitems}

\index{background (experimentdb.reagents.models.Strain attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Strain.background}\pysigline{\code{Strain.}\bfcode{background}}{}
\end{fulllineitems}

\index{experiment\_set (experimentdb.reagents.models.Strain attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Strain.experiment_set}\pysigline{\code{Strain.}\bfcode{experiment\_set}}{}
\end{fulllineitems}

\index{get\_absolute\_url() (experimentdb.reagents.models.Strain method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Strain.get_absolute_url}\pysiglinewithargsret{\code{Strain.}\bfcode{get\_absolute\_url}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{get\_location\_display() (experimentdb.reagents.models.Strain method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Strain.get_location_display}\pysiglinewithargsret{\code{Strain.}\bfcode{get\_location\_display}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{get\_species\_display() (experimentdb.reagents.models.Strain method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Strain.get_species_display}\pysiglinewithargsret{\code{Strain.}\bfcode{get\_species\_display}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{plasmids (experimentdb.reagents.models.Strain attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Strain.plasmids}\pysigline{\code{Strain.}\bfcode{plasmids}}{}
\end{fulllineitems}

\index{protein (experimentdb.reagents.models.Strain attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Strain.protein}\pysigline{\code{Strain.}\bfcode{protein}}{}
\end{fulllineitems}

\index{reference (experimentdb.reagents.models.Strain attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Strain.reference}\pysigline{\code{Strain.}\bfcode{reference}}{}
\end{fulllineitems}

\index{researcher (experimentdb.reagents.models.Strain attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Strain.researcher}\pysigline{\code{Strain.}\bfcode{researcher}}{}
\end{fulllineitems}

\index{save() (experimentdb.reagents.models.Strain method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Strain.save}\pysiglinewithargsret{\code{Strain.}\bfcode{save}}{}{}
The save is over-ridden to slugify the name field into a slugfield.


\end{fulllineitems}

\index{selection (experimentdb.reagents.models.Strain attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Strain.selection}\pysigline{\code{Strain.}\bfcode{selection}}{}
\end{fulllineitems}

\index{strain\_set (experimentdb.reagents.models.Strain attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Strain.strain_set}\pysigline{\code{Strain.}\bfcode{strain\_set}}{}
\end{fulllineitems}

\index{strain\_species (experimentdb.reagents.models.Strain attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Strain.strain_species}\pysigline{\code{Strain.}\bfcode{strain\_species}}{}
\end{fulllineitems}

\index{vendor (experimentdb.reagents.models.Strain attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.models.Strain.vendor}\pysigline{\code{Strain.}\bfcode{vendor}}{}
\end{fulllineitems}


\end{fulllineitems}



\subsection{Views}
\label{api:id23}\phantomsection\label{api:module-experimentdb.reagents.views}\index{experimentdb.reagents.views (module)}\index{antibody\_lookup() (in module experimentdb.reagents.views)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.views.antibody_lookup}\pysiglinewithargsret{\code{experimentdb.reagents.views.}\bfcode{antibody\_lookup}}{\emph{request}}{}
A json lookup view for antibodies.


This view takes a get query item and returns a json dictionary of antibody objects matching that name


\end{fulllineitems}

\index{index() (in module experimentdb.reagents.views)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.views.index}\pysiglinewithargsret{\code{experimentdb.reagents.views.}\bfcode{index}}{\emph{request}, \emph{*args}, \emph{**kwargs}}{}
\end{fulllineitems}



\subsection{Lookups}
\label{api:id24}\phantomsection\label{api:module-experimentdb.reagents.lookups}\index{experimentdb.reagents.lookups (module)}
This is a configuration file for the ajax lookups for the reagents app.


See \href{http://code.google.com/p/django-ajax-selects/}{http://code.google.com/p/django-ajax-selects/} for information about configuring the ajax lookups.

\index{AntibodyLookup (class in experimentdb.reagents.lookups)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.lookups.AntibodyLookup}\pysigline{\strong{class }\code{experimentdb.reagents.lookups.}\bfcode{AntibodyLookup}}{}
This is the generic lookup for antibodies.


It is to be used for all antibody requests and directs to the `antibody' channel.

\index{format\_item() (experimentdb.reagents.lookups.AntibodyLookup method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.lookups.AntibodyLookup.format_item}\pysiglinewithargsret{\bfcode{format\_item}}{\emph{antibody}}{}
the display of a currently selected object in the area below the search box. html is OK


\end{fulllineitems}

\index{format\_result() (experimentdb.reagents.lookups.AntibodyLookup method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.lookups.AntibodyLookup.format_result}\pysiglinewithargsret{\bfcode{format\_result}}{\emph{antibody}}{}
This controls the display of the dropdown menu.


This is set to show the unicode name of the antibody, as well as the vendor and the source species.


\end{fulllineitems}

\index{get\_objects() (experimentdb.reagents.lookups.AntibodyLookup method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.lookups.AntibodyLookup.get_objects}\pysiglinewithargsret{\bfcode{get\_objects}}{\emph{ids}}{}
given a list of ids, return the objects ordered as you would like them on the admin page.
this is for displaying the currently selected items (in the case of a ManyToMany field)


\end{fulllineitems}

\index{get\_query() (experimentdb.reagents.lookups.AntibodyLookup method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.lookups.AntibodyLookup.get_query}\pysiglinewithargsret{\bfcode{get\_query}}{\emph{q}, \emph{request}}{}
This sets up the query for the lookup.


The lookup searches the name of the antibody.


\end{fulllineitems}


\end{fulllineitems}

\index{CellLineLookup (class in experimentdb.reagents.lookups)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.lookups.CellLineLookup}\pysigline{\strong{class }\code{experimentdb.reagents.lookups.}\bfcode{CellLineLookup}}{}
This is the generic lookup for strains.


It is to be used for all cell line requests and directs to the `cell' channel.

\index{format\_item() (experimentdb.reagents.lookups.CellLineLookup method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.lookups.CellLineLookup.format_item}\pysiglinewithargsret{\bfcode{format\_item}}{\emph{cell}}{}
the display of a currently selected object in the area below the search box. html is OK


\end{fulllineitems}

\index{format\_result() (experimentdb.reagents.lookups.CellLineLookup method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.lookups.CellLineLookup.format_result}\pysiglinewithargsret{\bfcode{format\_result}}{\emph{cell}}{}
This controls the display of the dropdown menu.


This is set to show the unicode name of the cell line.


\end{fulllineitems}

\index{get\_objects() (experimentdb.reagents.lookups.CellLineLookup method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.lookups.CellLineLookup.get_objects}\pysiglinewithargsret{\bfcode{get\_objects}}{\emph{ids}}{}
given a list of ids, return the objects ordered as you would like them on the admin page.
this is for displaying the currently selected items (in the case of a ManyToMany field)


\end{fulllineitems}

\index{get\_query() (experimentdb.reagents.lookups.CellLineLookup method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.lookups.CellLineLookup.get_query}\pysiglinewithargsret{\bfcode{get\_query}}{\emph{q}, \emph{request}}{}
This sets up the query for the lookup.


The lookup searches the name of the cell.


\end{fulllineitems}


\end{fulllineitems}

\index{ChemicalLookup (class in experimentdb.reagents.lookups)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.lookups.ChemicalLookup}\pysigline{\strong{class }\code{experimentdb.reagents.lookups.}\bfcode{ChemicalLookup}}{}
This is the generic lookup for chemicals.


It is to be used for all chemical requests and directs to the `chemical' channel.

\index{format\_item() (experimentdb.reagents.lookups.ChemicalLookup method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.lookups.ChemicalLookup.format_item}\pysiglinewithargsret{\bfcode{format\_item}}{\emph{chemical}}{}
the display of a currently selected object in the area below the search box. html is OK


\end{fulllineitems}

\index{format\_result() (experimentdb.reagents.lookups.ChemicalLookup method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.lookups.ChemicalLookup.format_result}\pysiglinewithargsret{\bfcode{format\_result}}{\emph{chemical}}{}
This controls the display of the dropdown menu.


This is set to show the unicode name of the chemical.


\end{fulllineitems}

\index{get\_objects() (experimentdb.reagents.lookups.ChemicalLookup method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.lookups.ChemicalLookup.get_objects}\pysiglinewithargsret{\bfcode{get\_objects}}{\emph{ids}}{}
given a list of ids, return the objects ordered as you would like them on the admin page.
this is for displaying the currently selected items (in the case of a ManyToMany field)


\end{fulllineitems}

\index{get\_query() (experimentdb.reagents.lookups.ChemicalLookup method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.lookups.ChemicalLookup.get_query}\pysiglinewithargsret{\bfcode{get\_query}}{\emph{q}, \emph{request}}{}
This sets up the query for the lookup.


The lookup searches the name of the chemical.


\end{fulllineitems}


\end{fulllineitems}

\index{ConstructLookup (class in experimentdb.reagents.lookups)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.lookups.ConstructLookup}\pysigline{\strong{class }\code{experimentdb.reagents.lookups.}\bfcode{ConstructLookup}}{}
This is the generic lookup for constructs.


It is to be used for all construct requests and directs to the `construct' channel.

\index{format\_item() (experimentdb.reagents.lookups.ConstructLookup method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.lookups.ConstructLookup.format_item}\pysiglinewithargsret{\bfcode{format\_item}}{\emph{construct}}{}
the display of a currently selected object in the area below the search box. html is OK


\end{fulllineitems}

\index{format\_result() (experimentdb.reagents.lookups.ConstructLookup method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.lookups.ConstructLookup.format_result}\pysiglinewithargsret{\bfcode{format\_result}}{\emph{construct}}{}
This controls the display of the dropdown menu.


This is set to show the unicode name of the construct.


\end{fulllineitems}

\index{get\_objects() (experimentdb.reagents.lookups.ConstructLookup method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.lookups.ConstructLookup.get_objects}\pysiglinewithargsret{\bfcode{get\_objects}}{\emph{ids}}{}
given a list of ids, return the objects ordered as you would like them on the admin page.
this is for displaying the currently selected items (in the case of a ManyToMany field)


\end{fulllineitems}

\index{get\_query() (experimentdb.reagents.lookups.ConstructLookup method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.lookups.ConstructLookup.get_query}\pysiglinewithargsret{\bfcode{get\_query}}{\emph{q}, \emph{request}}{}
This sets up the query for the lookup.


The lookup searches the name of the construct.


\end{fulllineitems}


\end{fulllineitems}

\index{SiRNALookup (class in experimentdb.reagents.lookups)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.lookups.SiRNALookup}\pysigline{\strong{class }\code{experimentdb.reagents.lookups.}\bfcode{SiRNALookup}}{}
This is the generic lookup for siRNA.


It is to be used for all siRNA requests and directs to the `siRNA' channel
This channel will \textbf{not} search for all Primer objects, only the ones with primer\_type=''siRNA''.

\index{format\_item() (experimentdb.reagents.lookups.SiRNALookup method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.lookups.SiRNALookup.format_item}\pysiglinewithargsret{\bfcode{format\_item}}{\emph{siRNA}}{}
the display of a currently selected object in the area below the search box. html is OK


\end{fulllineitems}

\index{format\_result() (experimentdb.reagents.lookups.SiRNALookup method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.lookups.SiRNALookup.format_result}\pysiglinewithargsret{\bfcode{format\_result}}{\emph{siRNA}}{}
This controls the display of the dropdown menu.


This is set to show the unicode name of the siRNA line.


\end{fulllineitems}

\index{get\_objects() (experimentdb.reagents.lookups.SiRNALookup method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.lookups.SiRNALookup.get_objects}\pysiglinewithargsret{\bfcode{get\_objects}}{\emph{ids}}{}
given a list of ids, return the objects ordered as you would like them on the admin page.
this is for displaying the currently selected items (in the case of a ManyToMany field)


\end{fulllineitems}

\index{get\_query() (experimentdb.reagents.lookups.SiRNALookup method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.lookups.SiRNALookup.get_query}\pysiglinewithargsret{\bfcode{get\_query}}{\emph{q}, \emph{request}}{}
This sets up the query for the lookup.


The lookup searches the name of the siRNA.


\end{fulllineitems}


\end{fulllineitems}

\index{StrainLookup (class in experimentdb.reagents.lookups)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.lookups.StrainLookup}\pysigline{\strong{class }\code{experimentdb.reagents.lookups.}\bfcode{StrainLookup}}{}
This is the generic lookup for strains.


It is to be used for all strain requests and directs to the `strain' channel.

\index{format\_item() (experimentdb.reagents.lookups.StrainLookup method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.lookups.StrainLookup.format_item}\pysiglinewithargsret{\bfcode{format\_item}}{\emph{strain}}{}
the display of a currently selected object in the area below the search box. html is OK


\end{fulllineitems}

\index{format\_result() (experimentdb.reagents.lookups.StrainLookup method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.lookups.StrainLookup.format_result}\pysiglinewithargsret{\bfcode{format\_result}}{\emph{strain}}{}
This controls the display of the dropdown menu.


This is set to show the unicode name of the strain.


\end{fulllineitems}

\index{get\_objects() (experimentdb.reagents.lookups.StrainLookup method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.lookups.StrainLookup.get_objects}\pysiglinewithargsret{\bfcode{get\_objects}}{\emph{ids}}{}
given a list of ids, return the objects ordered as you would like them on the admin page.
this is for displaying the currently selected items (in the case of a ManyToMany field)


\end{fulllineitems}

\index{get\_query() (experimentdb.reagents.lookups.StrainLookup method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.lookups.StrainLookup.get_query}\pysiglinewithargsret{\bfcode{get\_query}}{\emph{q}, \emph{request}}{}
This sets up the query for the lookup.


The lookup searches the name of the strain.


\end{fulllineitems}


\end{fulllineitems}



\subsection{URLconfs}
\label{api:id25}\phantomsection\label{api:module-experimentdb.reagents.urls}\index{experimentdb.reagents.urls (module)}
URLconfs for reagent models.


In general these urls have the names model-list, model-detail, model-new, model-edit and model-delete.



\subsection{Tests}
\label{api:id26}\phantomsection\label{api:module-experimentdb.reagents.tests}\index{experimentdb.reagents.tests (module)}
This file contains tests for the reagents application.


These tests include model and view tests for Strain, Primer, Cell, Antibody, Construct, Chemical, Species and Selection objects.

\index{AntibodyModelTests (class in experimentdb.reagents.tests)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.AntibodyModelTests}\pysiglinewithargsret{\strong{class }\code{experimentdb.reagents.tests.}\bfcode{AntibodyModelTests}}{\emph{methodName='runTest'}}{}
Tests the model attributes of Antibody objects contained in the reagents app.

\index{setUp() (experimentdb.reagents.tests.AntibodyModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.AntibodyModelTests.setUp}\pysiglinewithargsret{\bfcode{setUp}}{}{}
Instantiate the test client.


\end{fulllineitems}

\index{tearDown() (experimentdb.reagents.tests.AntibodyModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.AntibodyModelTests.tearDown}\pysiglinewithargsret{\bfcode{tearDown}}{}{}
Depopulate created model instances from test database.


\end{fulllineitems}

\index{test\_antibody\_slugify() (experimentdb.reagents.tests.AntibodyModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.AntibodyModelTests.test_antibody_slugify}\pysiglinewithargsret{\bfcode{test\_antibody\_slugify}}{}{}
This is a test for the antibody name being correctly slugified


\end{fulllineitems}

\index{test\_create\_antibody\_all\_fields() (experimentdb.reagents.tests.AntibodyModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.AntibodyModelTests.test_create_antibody_all_fields}\pysiglinewithargsret{\bfcode{test\_create\_antibody\_all\_fields}}{}{}
This is a test for creating a new antibody object, with only the all fields being entered


\end{fulllineitems}

\index{test\_create\_antibody\_minimal() (experimentdb.reagents.tests.AntibodyModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.AntibodyModelTests.test_create_antibody_minimal}\pysiglinewithargsret{\bfcode{test\_create\_antibody\_minimal}}{}{}
This is a test for creating a new antibody object, with only the minimum fields being entered


\end{fulllineitems}


\end{fulllineitems}

\index{CellModelTests (class in experimentdb.reagents.tests)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.CellModelTests}\pysiglinewithargsret{\strong{class }\code{experimentdb.reagents.tests.}\bfcode{CellModelTests}}{\emph{methodName='runTest'}}{}
Tests the model attributes of Cell objects contained in the reagents app.

\index{setUp() (experimentdb.reagents.tests.CellModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.CellModelTests.setUp}\pysiglinewithargsret{\bfcode{setUp}}{}{}
Instantiate the test client.


\end{fulllineitems}

\index{tearDown() (experimentdb.reagents.tests.CellModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.CellModelTests.tearDown}\pysiglinewithargsret{\bfcode{tearDown}}{}{}
Depopulate created model instances from test database.


\end{fulllineitems}

\index{test\_cell\_line\_slugify() (experimentdb.reagents.tests.CellModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.CellModelTests.test_cell_line_slugify}\pysiglinewithargsret{\bfcode{test\_cell\_line\_slugify}}{}{}
This is a test for the cell line name being correctly slugified


\end{fulllineitems}

\index{test\_create\_cell\_line\_all\_fields() (experimentdb.reagents.tests.CellModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.CellModelTests.test_create_cell_line_all_fields}\pysiglinewithargsret{\bfcode{test\_create\_cell\_line\_all\_fields}}{}{}
This is a test for creating a new cell\_line object, with only the all fields being entered


\end{fulllineitems}

\index{test\_create\_cell\_line\_minimal() (experimentdb.reagents.tests.CellModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.CellModelTests.test_create_cell_line_minimal}\pysiglinewithargsret{\bfcode{test\_create\_cell\_line\_minimal}}{}{}
This is a test for creating a new cell line object, with only the minimum fields being entered


\end{fulllineitems}


\end{fulllineitems}

\index{ChemicalModelTests (class in experimentdb.reagents.tests)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.ChemicalModelTests}\pysiglinewithargsret{\strong{class }\code{experimentdb.reagents.tests.}\bfcode{ChemicalModelTests}}{\emph{methodName='runTest'}}{}
Tests the model attributes of Chemical objects contained in the reagents app.

\index{setUp() (experimentdb.reagents.tests.ChemicalModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.ChemicalModelTests.setUp}\pysiglinewithargsret{\bfcode{setUp}}{}{}
Instantiate the test client.


\end{fulllineitems}

\index{tearDown() (experimentdb.reagents.tests.ChemicalModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.ChemicalModelTests.tearDown}\pysiglinewithargsret{\bfcode{tearDown}}{}{}
Depopulate created model instances from test database.


\end{fulllineitems}

\index{test\_chemical\_slugify() (experimentdb.reagents.tests.ChemicalModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.ChemicalModelTests.test_chemical_slugify}\pysiglinewithargsret{\bfcode{test\_chemical\_slugify}}{}{}
This is a test for the cell line name being correctly slugified


\end{fulllineitems}

\index{test\_create\_chemical\_all\_fields() (experimentdb.reagents.tests.ChemicalModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.ChemicalModelTests.test_create_chemical_all_fields}\pysiglinewithargsret{\bfcode{test\_create\_chemical\_all\_fields}}{}{}
This is a test for creating a new chemical object, with only the all fields being entered


\end{fulllineitems}

\index{test\_create\_chemical\_minimal() (experimentdb.reagents.tests.ChemicalModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.ChemicalModelTests.test_create_chemical_minimal}\pysiglinewithargsret{\bfcode{test\_create\_chemical\_minimal}}{}{}
This is a test for creating a new chemical object, with only the minimum fields being entered


\end{fulllineitems}


\end{fulllineitems}

\index{ConstructModelTests (class in experimentdb.reagents.tests)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.ConstructModelTests}\pysiglinewithargsret{\strong{class }\code{experimentdb.reagents.tests.}\bfcode{ConstructModelTests}}{\emph{methodName='runTest'}}{}
Tests the model attributes of Construct objects contained in the reagents app.

\index{setUp() (experimentdb.reagents.tests.ConstructModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.ConstructModelTests.setUp}\pysiglinewithargsret{\bfcode{setUp}}{}{}
Instantiate the test client.


\end{fulllineitems}

\index{tearDown() (experimentdb.reagents.tests.ConstructModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.ConstructModelTests.tearDown}\pysiglinewithargsret{\bfcode{tearDown}}{}{}
Depopulate created model instances from test database.


\end{fulllineitems}

\index{test\_construct\_slugify() (experimentdb.reagents.tests.ConstructModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.ConstructModelTests.test_construct_slugify}\pysiglinewithargsret{\bfcode{test\_construct\_slugify}}{}{}
This is a test for the construct name being correctly slugified


\end{fulllineitems}

\index{test\_create\_cell\_line\_minimal() (experimentdb.reagents.tests.ConstructModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.ConstructModelTests.test_create_cell_line_minimal}\pysiglinewithargsret{\bfcode{test\_create\_cell\_line\_minimal}}{}{}
This is a test for creating a new construct object, with only the minimum fields being entered


\end{fulllineitems}

\index{test\_create\_construct\_all\_fields() (experimentdb.reagents.tests.ConstructModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.ConstructModelTests.test_create_construct_all_fields}\pysiglinewithargsret{\bfcode{test\_create\_construct\_all\_fields}}{}{}
This is a test for creating a new construct object, with only the all fields being entered


\end{fulllineitems}


\end{fulllineitems}

\index{PrimerModelTests (class in experimentdb.reagents.tests)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.PrimerModelTests}\pysiglinewithargsret{\strong{class }\code{experimentdb.reagents.tests.}\bfcode{PrimerModelTests}}{\emph{methodName='runTest'}}{}
Tests the model attributes of Primer objects contained in the reagents app.

\index{setUp() (experimentdb.reagents.tests.PrimerModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.PrimerModelTests.setUp}\pysiglinewithargsret{\bfcode{setUp}}{}{}
Instantiate the test client.


\end{fulllineitems}

\index{tearDown() (experimentdb.reagents.tests.PrimerModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.PrimerModelTests.tearDown}\pysiglinewithargsret{\bfcode{tearDown}}{}{}
Depopulate created model instances from test database.


\end{fulllineitems}

\index{test\_create\_primer\_all\_fields() (experimentdb.reagents.tests.PrimerModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.PrimerModelTests.test_create_primer_all_fields}\pysiglinewithargsret{\bfcode{test\_create\_primer\_all\_fields}}{}{}
This is a test for creating a new primer object, with only the all fields being entered


\end{fulllineitems}

\index{test\_create\_primer\_minimal() (experimentdb.reagents.tests.PrimerModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.PrimerModelTests.test_create_primer_minimal}\pysiglinewithargsret{\bfcode{test\_create\_primer\_minimal}}{}{}
This is a test for creating a new primer object, with only the minimum fields being entered


\end{fulllineitems}

\index{test\_primer\_slugify() (experimentdb.reagents.tests.PrimerModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.PrimerModelTests.test_primer_slugify}\pysiglinewithargsret{\bfcode{test\_primer\_slugify}}{}{}
This is a test for the primer name being correctly slugified


\end{fulllineitems}


\end{fulllineitems}

\index{SelectionModelTests (class in experimentdb.reagents.tests)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.SelectionModelTests}\pysiglinewithargsret{\strong{class }\code{experimentdb.reagents.tests.}\bfcode{SelectionModelTests}}{\emph{methodName='runTest'}}{}
Tests the model attributes of Selection objects contained in the reagents app.

\index{setUp() (experimentdb.reagents.tests.SelectionModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.SelectionModelTests.setUp}\pysiglinewithargsret{\bfcode{setUp}}{}{}
Instantiate the test client.


\end{fulllineitems}

\index{tearDown() (experimentdb.reagents.tests.SelectionModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.SelectionModelTests.tearDown}\pysiglinewithargsret{\bfcode{tearDown}}{}{}
Depopulate created model instances from test database.


\end{fulllineitems}

\index{test\_create\_selection\_all\_fields() (experimentdb.reagents.tests.SelectionModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.SelectionModelTests.test_create_selection_all_fields}\pysiglinewithargsret{\bfcode{test\_create\_selection\_all\_fields}}{}{}
This is a test for creating a new selection object, with only the all fields being entered


\end{fulllineitems}

\index{test\_create\_selection\_minimal() (experimentdb.reagents.tests.SelectionModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.SelectionModelTests.test_create_selection_minimal}\pysiglinewithargsret{\bfcode{test\_create\_selection\_minimal}}{}{}
This is a test for creating a new selection object, with only the minimum fields being entered


\end{fulllineitems}

\index{test\_selection\_slugify() (experimentdb.reagents.tests.SelectionModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.SelectionModelTests.test_selection_slugify}\pysiglinewithargsret{\bfcode{test\_selection\_slugify}}{}{}
This is a test for the cell line name being correctly slugified


\end{fulllineitems}


\end{fulllineitems}

\index{SpeciesModelTests (class in experimentdb.reagents.tests)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.SpeciesModelTests}\pysiglinewithargsret{\strong{class }\code{experimentdb.reagents.tests.}\bfcode{SpeciesModelTests}}{\emph{methodName='runTest'}}{}
Tests the model attributes of Species objects contained in the reagents app.

\index{setUp() (experimentdb.reagents.tests.SpeciesModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.SpeciesModelTests.setUp}\pysiglinewithargsret{\bfcode{setUp}}{}{}
Instantiate the test client.


\end{fulllineitems}

\index{tearDown() (experimentdb.reagents.tests.SpeciesModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.SpeciesModelTests.tearDown}\pysiglinewithargsret{\bfcode{tearDown}}{}{}
Depopulate created model instances from test database.


\end{fulllineitems}

\index{test\_create\_species\_all\_fields() (experimentdb.reagents.tests.SpeciesModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.SpeciesModelTests.test_create_species_all_fields}\pysiglinewithargsret{\bfcode{test\_create\_species\_all\_fields}}{}{}
This is a test for creating a new species object, with only the all fields being entered


\end{fulllineitems}

\index{test\_create\_species\_minimal() (experimentdb.reagents.tests.SpeciesModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.SpeciesModelTests.test_create_species_minimal}\pysiglinewithargsret{\bfcode{test\_create\_species\_minimal}}{}{}
This is a test for creating a new species object, with only the minimum fields being entered


\end{fulllineitems}

\index{test\_species\_slugify() (experimentdb.reagents.tests.SpeciesModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.SpeciesModelTests.test_species_slugify}\pysiglinewithargsret{\bfcode{test\_species\_slugify}}{}{}
This is a test for the cell line name being correctly slugified


\end{fulllineitems}


\end{fulllineitems}

\index{StrainModelTests (class in experimentdb.reagents.tests)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.StrainModelTests}\pysiglinewithargsret{\strong{class }\code{experimentdb.reagents.tests.}\bfcode{StrainModelTests}}{\emph{methodName='runTest'}}{}
Tests the model attributes of Strain objects contained in the reagents app.

\index{setUp() (experimentdb.reagents.tests.StrainModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.StrainModelTests.setUp}\pysiglinewithargsret{\bfcode{setUp}}{}{}
Instantiate the test client.


\end{fulllineitems}

\index{tearDown() (experimentdb.reagents.tests.StrainModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.StrainModelTests.tearDown}\pysiglinewithargsret{\bfcode{tearDown}}{}{}
Depopulate created model instances from test database.


\end{fulllineitems}

\index{test\_create\_strain\_all\_fields() (experimentdb.reagents.tests.StrainModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.StrainModelTests.test_create_strain_all_fields}\pysiglinewithargsret{\bfcode{test\_create\_strain\_all\_fields}}{}{}
This is a test for creating a new strain object, with only the all fields being entered


\end{fulllineitems}

\index{test\_create\_strain\_minimal() (experimentdb.reagents.tests.StrainModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.StrainModelTests.test_create_strain_minimal}\pysiglinewithargsret{\bfcode{test\_create\_strain\_minimal}}{}{}
This is a test for creating a new strain object, with only the minimum fields being entered


\end{fulllineitems}

\index{test\_strain\_slugify() (experimentdb.reagents.tests.StrainModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.reagents.tests.StrainModelTests.test_strain_slugify}\pysiglinewithargsret{\bfcode{test\_strain\_slugify}}{}{}
This is a test for the cell line name being correctly slugified


\end{fulllineitems}


\end{fulllineitems}



\subsection{Fixtures}
\label{api:id27}\phantomsection\label{api:module-experimentdb.reagents.fixtures}\index{experimentdb.reagents.fixtures (module)}
This folder contains fixtures for the reagents app.


Currently there are the following text fixtures
- Species model named test\_species.json
- Selection model named test\_selection.json
- Strain model named test\_strain.json
- Construct model named test\_construct.json



\section{Sharing Package}
\label{api:sharing-package}\label{api:module-experimentdb.sharing}\index{experimentdb.sharing (module)}

\subsection{Models}
\label{api:id28}\phantomsection\label{api:module-experimentdb.sharing.models}\index{experimentdb.sharing.models (module)}
This package defines the database models for for the sharing application.


This application tracks shipments of constructs to other groups.


These tests include the following models:
- Institution
- Laboratory
- Recipient
- ConstructShipment


In the terms of this application, \textbf{ConstructShipments} are sent to \textbf{Recipients}, who are in \textbf{Laboratories} at \textbf{Institutions}.

\index{ConstructShipment (class in experimentdb.sharing.models)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.models.ConstructShipment}\pysiglinewithargsret{\strong{class }\code{experimentdb.sharing.models.}\bfcode{ConstructShipment}}{\emph{*args}, \emph{**kwargs}}{}
This class describes a shipment of constructs.


The required fields are \textbf{constructs}, \textbf{ship\_date}, \textbf{recipient} (who is defined as part of a Laboratory and in turn an Institution).

\index{ConstructShipment.DoesNotExist}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.models.ConstructShipment.DoesNotExist}\pysigline{\strong{exception }\bfcode{DoesNotExist}}{}
\end{fulllineitems}

\index{ConstructShipment.MultipleObjectsReturned}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.models.ConstructShipment.MultipleObjectsReturned}\pysigline{\strong{exception }\code{ConstructShipment.}\bfcode{MultipleObjectsReturned}}{}
\end{fulllineitems}

\index{constructs (experimentdb.sharing.models.ConstructShipment attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.models.ConstructShipment.constructs}\pysigline{\code{ConstructShipment.}\bfcode{constructs}}{}
\end{fulllineitems}

\index{get\_absolute\_url() (experimentdb.sharing.models.ConstructShipment method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.models.ConstructShipment.get_absolute_url}\pysiglinewithargsret{\code{ConstructShipment.}\bfcode{get\_absolute\_url}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{get\_next\_by\_ship\_date() (experimentdb.sharing.models.ConstructShipment method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.models.ConstructShipment.get_next_by_ship_date}\pysiglinewithargsret{\code{ConstructShipment.}\bfcode{get\_next\_by\_ship\_date}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{get\_previous\_by\_ship\_date() (experimentdb.sharing.models.ConstructShipment method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.models.ConstructShipment.get_previous_by_ship_date}\pysiglinewithargsret{\code{ConstructShipment.}\bfcode{get\_previous\_by\_ship\_date}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{recipient (experimentdb.sharing.models.ConstructShipment attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.models.ConstructShipment.recipient}\pysigline{\code{ConstructShipment.}\bfcode{recipient}}{}
\end{fulllineitems}


\end{fulllineitems}

\index{Institution (class in experimentdb.sharing.models)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.models.Institution}\pysiglinewithargsret{\strong{class }\code{experimentdb.sharing.models.}\bfcode{Institution}}{\emph{*args}, \emph{**kwargs}}{}
This class defines Institution models.


The only required is \textbf{institution}.
The institution describes part of the address (city/state/country) the rest is defined under Laboratory.

\index{Institution.DoesNotExist}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.models.Institution.DoesNotExist}\pysigline{\strong{exception }\bfcode{DoesNotExist}}{}
\end{fulllineitems}

\index{Institution.MultipleObjectsReturned}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.models.Institution.MultipleObjectsReturned}\pysigline{\strong{exception }\code{Institution.}\bfcode{MultipleObjectsReturned}}{}
\end{fulllineitems}

\index{get\_country\_display() (experimentdb.sharing.models.Institution method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.models.Institution.get_country_display}\pysiglinewithargsret{\code{Institution.}\bfcode{get\_country\_display}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{get\_institution\_type\_display() (experimentdb.sharing.models.Institution method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.models.Institution.get_institution_type_display}\pysiglinewithargsret{\code{Institution.}\bfcode{get\_institution\_type\_display}}{\emph{*moreargs}, \emph{**morekwargs}}{}
\end{fulllineitems}

\index{laboratory\_set (experimentdb.sharing.models.Institution attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.models.Institution.laboratory_set}\pysigline{\code{Institution.}\bfcode{laboratory\_set}}{}
\end{fulllineitems}


\end{fulllineitems}

\index{Laboratory (class in experimentdb.sharing.models)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.models.Laboratory}\pysiglinewithargsret{\strong{class }\code{experimentdb.sharing.models.}\bfcode{Laboratory}}{\emph{*args}, \emph{**kwargs}}{}
This class describes groups or laboratories.


This class has two required fields, \textbf{principal\_investigator} and \textbf{institution}.
In this context, a laboratory could be a single person or a group of people at an institution.
Typically the recipient of the shipment works at the laboratory.
The laboratory may or may not also be a contact, as defined in the external app.

\index{Laboratory.DoesNotExist}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.models.Laboratory.DoesNotExist}\pysigline{\strong{exception }\bfcode{DoesNotExist}}{}
\end{fulllineitems}

\index{Laboratory.MultipleObjectsReturned}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.models.Laboratory.MultipleObjectsReturned}\pysigline{\strong{exception }\code{Laboratory.}\bfcode{MultipleObjectsReturned}}{}
\end{fulllineitems}

\index{contact (experimentdb.sharing.models.Laboratory attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.models.Laboratory.contact}\pysigline{\code{Laboratory.}\bfcode{contact}}{}
\end{fulllineitems}

\index{institution (experimentdb.sharing.models.Laboratory attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.models.Laboratory.institution}\pysigline{\code{Laboratory.}\bfcode{institution}}{}
\end{fulllineitems}

\index{recipient\_set (experimentdb.sharing.models.Laboratory attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.models.Laboratory.recipient_set}\pysigline{\code{Laboratory.}\bfcode{recipient\_set}}{}
\end{fulllineitems}


\end{fulllineitems}

\index{Recipient (class in experimentdb.sharing.models)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.models.Recipient}\pysiglinewithargsret{\strong{class }\code{experimentdb.sharing.models.}\bfcode{Recipient}}{\emph{*args}, \emph{**kwargs}}{}
This class describes the recipient of a shipment.


The recipient could be the principal investigator, or a member of their group.
The required fields for this model are \textbf{first\_name}, \textbf{last\_name} and \textbf{lab}.

\index{Recipient.DoesNotExist}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.models.Recipient.DoesNotExist}\pysigline{\strong{exception }\bfcode{DoesNotExist}}{}
\end{fulllineitems}

\index{Recipient.MultipleObjectsReturned}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.models.Recipient.MultipleObjectsReturned}\pysigline{\strong{exception }\code{Recipient.}\bfcode{MultipleObjectsReturned}}{}
\end{fulllineitems}

\index{constructshipment\_set (experimentdb.sharing.models.Recipient attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.models.Recipient.constructshipment_set}\pysigline{\code{Recipient.}\bfcode{constructshipment\_set}}{}
\end{fulllineitems}

\index{lab (experimentdb.sharing.models.Recipient attribute)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.models.Recipient.lab}\pysigline{\code{Recipient.}\bfcode{lab}}{}
\end{fulllineitems}


\end{fulllineitems}



\subsection{Views}
\label{api:id29}\phantomsection\label{api:module-experimentdb.sharing.views}\index{experimentdb.sharing.views (module)}

\subsection{URLconfs}
\label{api:id30}\phantomsection\label{api:module-experimentdb.sharing.urls}\index{experimentdb.sharing.urls (module)}\index{change\_shipment() (in module experimentdb.sharing.urls)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.urls.change_shipment}\pysiglinewithargsret{\code{experimentdb.sharing.urls.}\bfcode{change\_shipment}}{\emph{request}, \emph{*args}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{create\_shipment() (in module experimentdb.sharing.urls)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.urls.create_shipment}\pysiglinewithargsret{\code{experimentdb.sharing.urls.}\bfcode{create\_shipment}}{\emph{request}, \emph{*args}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{delete\_shipment() (in module experimentdb.sharing.urls)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.urls.delete_shipment}\pysiglinewithargsret{\code{experimentdb.sharing.urls.}\bfcode{delete\_shipment}}{\emph{request}, \emph{*args}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{shipment\_detail() (in module experimentdb.sharing.urls)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.urls.shipment_detail}\pysiglinewithargsret{\code{experimentdb.sharing.urls.}\bfcode{shipment\_detail}}{\emph{request}, \emph{*args}, \emph{**kwargs}}{}
\end{fulllineitems}

\index{shipment\_list() (in module experimentdb.sharing.urls)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.urls.shipment_list}\pysiglinewithargsret{\code{experimentdb.sharing.urls.}\bfcode{shipment\_list}}{\emph{request}, \emph{*args}, \emph{**kwargs}}{}
\end{fulllineitems}



\subsection{Tests}
\label{api:id31}\phantomsection\label{api:module-experimentdb.sharing.tests}\index{experimentdb.sharing.tests (module)}
This file contains tests for the sharing application.


These tests include model and view tests for the following models:
- Institution
- Laboratory
- Recipient
- ConstructShipment

\index{ConstructShipmentModelTests (class in experimentdb.sharing.tests)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.tests.ConstructShipmentModelTests}\pysiglinewithargsret{\strong{class }\code{experimentdb.sharing.tests.}\bfcode{ConstructShipmentModelTests}}{\emph{methodName='runTest'}}{}
Tests the model attributes of ConstructShipment objects contained in the reagents app.

\index{setUp() (experimentdb.sharing.tests.ConstructShipmentModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.tests.ConstructShipmentModelTests.setUp}\pysiglinewithargsret{\bfcode{setUp}}{}{}
Instantiate the test client.


\end{fulllineitems}

\index{tearDown() (experimentdb.sharing.tests.ConstructShipmentModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.tests.ConstructShipmentModelTests.tearDown}\pysiglinewithargsret{\bfcode{tearDown}}{}{}
Depopulate created model instances from test database.


\end{fulllineitems}

\index{test\_create\_construct\_shipment\_all\_fields() (experimentdb.sharing.tests.ConstructShipmentModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.tests.ConstructShipmentModelTests.test_create_construct_shipment_all_fields}\pysiglinewithargsret{\bfcode{test\_create\_construct\_shipment\_all\_fields}}{}{}
This is a test for creating a new construct shipment object, with all fields being entered


\end{fulllineitems}

\index{test\_create\_construct\_shipment\_minimal() (experimentdb.sharing.tests.ConstructShipmentModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.tests.ConstructShipmentModelTests.test_create_construct_shipment_minimal}\pysiglinewithargsret{\bfcode{test\_create\_construct\_shipment\_minimal}}{}{}
This is a test for creating a new construct shipment, with only the minimum fields being entered


\end{fulllineitems}


\end{fulllineitems}

\index{InstitutionModelTests (class in experimentdb.sharing.tests)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.tests.InstitutionModelTests}\pysiglinewithargsret{\strong{class }\code{experimentdb.sharing.tests.}\bfcode{InstitutionModelTests}}{\emph{methodName='runTest'}}{}
Tests the model attributes of Laboratory objects contained in the reagents app.

\index{setUp() (experimentdb.sharing.tests.InstitutionModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.tests.InstitutionModelTests.setUp}\pysiglinewithargsret{\bfcode{setUp}}{}{}
Instantiate the test client.


\end{fulllineitems}

\index{tearDown() (experimentdb.sharing.tests.InstitutionModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.tests.InstitutionModelTests.tearDown}\pysiglinewithargsret{\bfcode{tearDown}}{}{}
Depopulate created model instances from test database.


\end{fulllineitems}

\index{test\_create\_institution\_all\_fields() (experimentdb.sharing.tests.InstitutionModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.tests.InstitutionModelTests.test_create_institution_all_fields}\pysiglinewithargsret{\bfcode{test\_create\_institution\_all\_fields}}{}{}
This is a test for creating a new institution object, with all fields being entered


\end{fulllineitems}

\index{test\_create\_institution\_minimal() (experimentdb.sharing.tests.InstitutionModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.tests.InstitutionModelTests.test_create_institution_minimal}\pysiglinewithargsret{\bfcode{test\_create\_institution\_minimal}}{}{}
This is a test for creating a new institution, with only the minimum fields being entered


\end{fulllineitems}


\end{fulllineitems}

\index{LaboratoryModelTests (class in experimentdb.sharing.tests)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.tests.LaboratoryModelTests}\pysiglinewithargsret{\strong{class }\code{experimentdb.sharing.tests.}\bfcode{LaboratoryModelTests}}{\emph{methodName='runTest'}}{}
Tests the model attributes of Laboratory objects contained in the reagents app.

\index{setUp() (experimentdb.sharing.tests.LaboratoryModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.tests.LaboratoryModelTests.setUp}\pysiglinewithargsret{\bfcode{setUp}}{}{}
Instantiate the test client.


\end{fulllineitems}

\index{tearDown() (experimentdb.sharing.tests.LaboratoryModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.tests.LaboratoryModelTests.tearDown}\pysiglinewithargsret{\bfcode{tearDown}}{}{}
Depopulate created model instances from test database.


\end{fulllineitems}

\index{test\_create\_laboratory\_all\_fields() (experimentdb.sharing.tests.LaboratoryModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.tests.LaboratoryModelTests.test_create_laboratory_all_fields}\pysiglinewithargsret{\bfcode{test\_create\_laboratory\_all\_fields}}{}{}
This is a test for creating a new recipient object, with all fields being entered


\end{fulllineitems}

\index{test\_create\_laboratory\_minimal() (experimentdb.sharing.tests.LaboratoryModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.tests.LaboratoryModelTests.test_create_laboratory_minimal}\pysiglinewithargsret{\bfcode{test\_create\_laboratory\_minimal}}{}{}
This is a test for creating a new laboratory, with only the minimum fields being entered


\end{fulllineitems}


\end{fulllineitems}

\index{RecipientModelTests (class in experimentdb.sharing.tests)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.tests.RecipientModelTests}\pysiglinewithargsret{\strong{class }\code{experimentdb.sharing.tests.}\bfcode{RecipientModelTests}}{\emph{methodName='runTest'}}{}
Tests the model attributes of Recipient objects contained in the reagents app.

\index{setUp() (experimentdb.sharing.tests.RecipientModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.tests.RecipientModelTests.setUp}\pysiglinewithargsret{\bfcode{setUp}}{}{}
Instantiate the test client.


\end{fulllineitems}

\index{tearDown() (experimentdb.sharing.tests.RecipientModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.tests.RecipientModelTests.tearDown}\pysiglinewithargsret{\bfcode{tearDown}}{}{}
Depopulate created model instances from test database.


\end{fulllineitems}

\index{test\_create\_recipient\_all\_fields() (experimentdb.sharing.tests.RecipientModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.tests.RecipientModelTests.test_create_recipient_all_fields}\pysiglinewithargsret{\bfcode{test\_create\_recipient\_all\_fields}}{}{}
This is a test for creating a new recipient, with all fields being entered


\end{fulllineitems}

\index{test\_create\_recipient\_minimal() (experimentdb.sharing.tests.RecipientModelTests method)}

\begin{fulllineitems}
\phantomsection\label{api:experimentdb.sharing.tests.RecipientModelTests.test_create_recipient_minimal}\pysiglinewithargsret{\bfcode{test\_create\_recipient\_minimal}}{}{}
This is a test for creating a new recipient, with only the minimum fields being entered


\end{fulllineitems}


\end{fulllineitems}



\subsection{Fixtures}
\label{api:id32}\phantomsection\label{api:module-experimentdb.sharing.fixtures}\index{experimentdb.sharing.fixtures (module)}
This folder contains data fixtures for the sharing app.


Currently it contains the following test fixtures:
- an Institution object as test\_institution.json
- a Laboratory object as test\_laboratory.json
- a Recipient object as test\_recipient.json



\chapter{Indices and tables}
\label{index:indices-and-tables}\begin{itemize}
\item {} 
\emph{genindex}


\item {} 
\emph{modindex}


\item {} 
\emph{search}


\end{itemize}


\renewcommand{\indexname}{Python Module Index}
\begin{theindex}
\def\bigletter#1{{\Large\sffamily#1}\nopagebreak\vspace{1mm}}
\bigletter{e}
\item {\texttt{experimentdb}}, \pageref{api:module-experimentdb}
\item {\texttt{experimentdb.cloning}}, \pageref{api:module-experimentdb.cloning}
\item {\texttt{experimentdb.cloning.models}}, \pageref{api:module-experimentdb.cloning.models}
\item {\texttt{experimentdb.cloning.urls}}, \pageref{api:module-experimentdb.cloning.urls}
\item {\texttt{experimentdb.cloning.views}}, \pageref{api:module-experimentdb.cloning.views}
\item {\texttt{experimentdb.data}}, \pageref{api:module-experimentdb.data}
\item {\texttt{experimentdb.data.lookups}}, \pageref{api:module-experimentdb.data.lookups}
\item {\texttt{experimentdb.data.models}}, \pageref{api:module-experimentdb.data.models}
\item {\texttt{experimentdb.data.urls}}, \pageref{api:module-experimentdb.data.urls}
\item {\texttt{experimentdb.data.views}}, \pageref{api:module-experimentdb.data.views}
\item {\texttt{experimentdb.datasets}}, \pageref{api:module-experimentdb.datasets}
\item {\texttt{experimentdb.datasets.models}}, \pageref{api:module-experimentdb.datasets.models}
\item {\texttt{experimentdb.datasets.views}}, \pageref{api:module-experimentdb.datasets.views}
\item {\texttt{experimentdb.external}}, \pageref{api:module-experimentdb.external}
\item {\texttt{experimentdb.external.fixtures}}, \pageref{api:module-experimentdb.external.fixtures}
\item {\texttt{experimentdb.external.models}}, \pageref{api:module-experimentdb.external.models}
\item {\texttt{experimentdb.external.tests}}, \pageref{api:module-experimentdb.external.tests}
\item {\texttt{experimentdb.external.urls}}, \pageref{api:module-experimentdb.external.urls}
\item {\texttt{experimentdb.external.views}}, \pageref{api:module-experimentdb.external.views}
\item {\texttt{experimentdb.proteins}}, \pageref{api:module-experimentdb.proteins}
\item {\texttt{experimentdb.proteins.fixtures}}, \pageref{api:module-experimentdb.proteins.fixtures}
\item {\texttt{experimentdb.proteins.lookups}}, \pageref{api:module-experimentdb.proteins.lookups}
\item {\texttt{experimentdb.proteins.models}}, \pageref{api:module-experimentdb.proteins.models}
\item {\texttt{experimentdb.proteins.urls}}, \pageref{api:module-experimentdb.proteins.urls}
\item {\texttt{experimentdb.proteins.views}}, \pageref{api:module-experimentdb.proteins.views}
\item {\texttt{experimentdb.reagents}}, \pageref{api:module-experimentdb.reagents}
\item {\texttt{experimentdb.reagents.fixtures}}, \pageref{api:module-experimentdb.reagents.fixtures}
\item {\texttt{experimentdb.reagents.lookups}}, \pageref{api:module-experimentdb.reagents.lookups}
\item {\texttt{experimentdb.reagents.models}}, \pageref{api:module-experimentdb.reagents.models}
\item {\texttt{experimentdb.reagents.tests}}, \pageref{api:module-experimentdb.reagents.tests}
\item {\texttt{experimentdb.reagents.urls}}, \pageref{api:module-experimentdb.reagents.urls}
\item {\texttt{experimentdb.reagents.views}}, \pageref{api:module-experimentdb.reagents.views}
\item {\texttt{experimentdb.sharing}}, \pageref{api:module-experimentdb.sharing}
\item {\texttt{experimentdb.sharing.fixtures}}, \pageref{api:module-experimentdb.sharing.fixtures}
\item {\texttt{experimentdb.sharing.models}}, \pageref{api:module-experimentdb.sharing.models}
\item {\texttt{experimentdb.sharing.tests}}, \pageref{api:module-experimentdb.sharing.tests}
\item {\texttt{experimentdb.sharing.urls}}, \pageref{api:module-experimentdb.sharing.urls}
\item {\texttt{experimentdb.sharing.views}}, \pageref{api:module-experimentdb.sharing.views}
\end{theindex}

\renewcommand{\indexname}{Index}
\printindex
\end{document}
